{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "import pandas as pd\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "xs = pd.read_csv('../data/training_variants.csv',index_col=0)\n",
    "x = pd.read_pickle('../data/new_tokens.pk')\n",
    "y = pd.read_csv('../data/training_variants.csv',index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>processed</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "      <td>[cyclin-depend, kinas, cdk, regul, varieti, fu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>[abstract, background, non-smal, cell, lung, c...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>[abstract, background, non-smal, cell, lung, c...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "      <td>[recent, evid, demonstr, acquir, uniparent, di...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>[oncogen, mutat, monomer, casita, b-lineag, ly...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                text  \\\n",
       "0  Cyclin-dependent kinases (CDKs) regulate a var...   \n",
       "1   Abstract Background  Non-small cell lung canc...   \n",
       "2   Abstract Background  Non-small cell lung canc...   \n",
       "3  Recent evidence has demonstrated that acquired...   \n",
       "4  Oncogenic mutations in the monomeric Casitas B...   \n",
       "\n",
       "                                           processed  \n",
       "0  [cyclin-depend, kinas, cdk, regul, varieti, fu...  \n",
       "1  [abstract, background, non-smal, cell, lung, c...  \n",
       "2  [abstract, background, non-smal, cell, lung, c...  \n",
       "3  [recent, evid, demonstr, acquir, uniparent, di...  \n",
       "4  [oncogen, mutat, monomer, casita, b-lineag, ly...  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>processed</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "      <td>[cyclin-depend, kinas, cdk, regul, varieti, fu...</td>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>[abstract, background, non-smal, cell, lung, c...</td>\n",
       "      <td>CBL</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "      <td>[abstract, background, non-smal, cell, lung, c...</td>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "      <td>[recent, evid, demonstr, acquir, uniparent, di...</td>\n",
       "      <td>CBL</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "      <td>[oncogen, mutat, monomer, casita, b-lineag, ly...</td>\n",
       "      <td>CBL</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                text  \\\n",
       "0  Cyclin-dependent kinases (CDKs) regulate a var...   \n",
       "1   Abstract Background  Non-small cell lung canc...   \n",
       "2   Abstract Background  Non-small cell lung canc...   \n",
       "3  Recent evidence has demonstrated that acquired...   \n",
       "4  Oncogenic mutations in the monomeric Casitas B...   \n",
       "\n",
       "                                           processed    Gene  \\\n",
       "0  [cyclin-depend, kinas, cdk, regul, varieti, fu...  FAM58A   \n",
       "1  [abstract, background, non-smal, cell, lung, c...     CBL   \n",
       "2  [abstract, background, non-smal, cell, lung, c...     CBL   \n",
       "3  [recent, evid, demonstr, acquir, uniparent, di...     CBL   \n",
       "4  [oncogen, mutat, monomer, casita, b-lineag, ly...     CBL   \n",
       "\n",
       "              Variation  Class  \n",
       "0  Truncating Mutations      1  \n",
       "1                 W802*      2  \n",
       "2                 Q249E      2  \n",
       "3                 N454D      3  \n",
       "4                 L399V      4  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "X = pd.merge(x,xs,left_index=True,right_index=True)\n",
    "X.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['FAM58A', 'CBL', 'SHOC2', 'TERT', 'DICER1', 'PTPRT', 'RHEB', 'SHQ1',\n",
       "       'CCND2', 'RAD50', 'CCND3', 'RIT1', 'CCNE1', 'RYBP', 'TGFBR1',\n",
       "       'TGFBR2', 'MSH6', 'KMT2D', 'LATS1', 'PBRM1'], dtype=object)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "X.Gene.unique()[:20]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['abstract',\n",
       " 'background',\n",
       " 'non-smal',\n",
       " 'cell',\n",
       " 'lung',\n",
       " 'cancer',\n",
       " 'nsclc',\n",
       " 'heterogen',\n",
       " 'group',\n",
       " 'disord',\n",
       " 'number',\n",
       " 'genet',\n",
       " 'proteom',\n",
       " 'alter',\n",
       " 'c-cbl',\n",
       " 'ubiquitin',\n",
       " 'ligas',\n",
       " 'adaptor',\n",
       " 'molecul',\n",
       " 'import',\n",
       " 'normal',\n",
       " 'homeostasi',\n",
       " 'cancer',\n",
       " 'determin',\n",
       " 'genet',\n",
       " 'variat',\n",
       " 'c-cbl',\n",
       " 'relationship',\n",
       " 'receptor',\n",
       " 'tyrosin',\n",
       " 'kinas',\n",
       " 'egfr',\n",
       " 'met',\n",
       " 'function',\n",
       " 'nsclc',\n",
       " 'method',\n",
       " 'find',\n",
       " 'use',\n",
       " 'archiv',\n",
       " 'formalin-fix',\n",
       " 'paraffin',\n",
       " 'embed',\n",
       " 'ffpe',\n",
       " 'extract',\n",
       " 'genom',\n",
       " 'dna',\n",
       " 'show',\n",
       " 'c-cbl',\n",
       " 'mutat',\n",
       " 'occur',\n",
       " 'somat',\n",
       " 'fashion',\n",
       " 'lung',\n",
       " 'cancer',\n",
       " 'c-cbl',\n",
       " 'mutat',\n",
       " 'mutual',\n",
       " 'exclus',\n",
       " 'met',\n",
       " 'egfr',\n",
       " 'mutat',\n",
       " 'howev',\n",
       " 'independ',\n",
       " 'p53',\n",
       " 'kra',\n",
       " 'mutat',\n",
       " 'normal',\n",
       " 'tumor',\n",
       " 'pairwis',\n",
       " 'analysi',\n",
       " 'signific',\n",
       " 'loss',\n",
       " 'heterozygos',\n",
       " 'loh',\n",
       " 'c-cbl',\n",
       " 'locu',\n",
       " 'none',\n",
       " 'sampl',\n",
       " 'reveal',\n",
       " 'mutat',\n",
       " 'remain',\n",
       " 'copi',\n",
       " 'c-cbl',\n",
       " 'c-cbl',\n",
       " 'loh',\n",
       " 'also',\n",
       " 'posit',\n",
       " 'correl',\n",
       " 'egfr',\n",
       " 'met',\n",
       " 'mutat',\n",
       " 'observ',\n",
       " 'sampl',\n",
       " 'use',\n",
       " 'select',\n",
       " 'c-cbl',\n",
       " 'somat',\n",
       " 'mutat',\n",
       " 's80n',\n",
       " 'h94i',\n",
       " 'q249e',\n",
       " 'w802',\n",
       " 'obtain',\n",
       " 'caucasian',\n",
       " 'taiwanes',\n",
       " 'african-american',\n",
       " 'sampl',\n",
       " 'respect',\n",
       " 'transfect',\n",
       " 'nsclc',\n",
       " 'cell',\n",
       " 'line',\n",
       " 'increas',\n",
       " 'cell',\n",
       " 'viabil',\n",
       " 'cell',\n",
       " 'motil',\n",
       " 'conclus',\n",
       " 'take',\n",
       " 'overal',\n",
       " 'mutat',\n",
       " 'rate',\n",
       " 'c-cbl',\n",
       " 'combin',\n",
       " 'somat',\n",
       " 'missens',\n",
       " 'mutat',\n",
       " 'loh',\n",
       " 'clear',\n",
       " 'c-cbl',\n",
       " 'highli',\n",
       " 'mutat',\n",
       " 'lung',\n",
       " 'cancer',\n",
       " 'may',\n",
       " 'play',\n",
       " 'essenti',\n",
       " 'role',\n",
       " 'lung',\n",
       " 'tumorigenesi',\n",
       " 'metastasi',\n",
       " 'introduct',\n",
       " 'alon',\n",
       " 'year',\n",
       " 'approxim',\n",
       " '219',\n",
       " '400',\n",
       " 'peopl',\n",
       " 'diagnos',\n",
       " 'lung',\n",
       " 'cancer',\n",
       " '145',\n",
       " '000',\n",
       " 'succumb',\n",
       " 'diseas',\n",
       " 'number',\n",
       " 'roughli',\n",
       " 'equival',\n",
       " 'combin',\n",
       " 'mortal',\n",
       " 'rate',\n",
       " 'cancer',\n",
       " 'breast',\n",
       " 'prostat',\n",
       " 'colon',\n",
       " 'liver',\n",
       " 'kidney',\n",
       " 'melanoma',\n",
       " 'addit',\n",
       " 'prognosi',\n",
       " 'usual',\n",
       " 'poor',\n",
       " 'five-year',\n",
       " 'surviv',\n",
       " 'rate',\n",
       " 'less',\n",
       " 'also',\n",
       " 'signific',\n",
       " 'ethnic',\n",
       " 'differ',\n",
       " 'lung',\n",
       " 'cancer',\n",
       " 'outcom',\n",
       " 'wors',\n",
       " 'black',\n",
       " 'compar',\n",
       " 'white',\n",
       " 'gender',\n",
       " 'differ',\n",
       " 'also',\n",
       " 'strike',\n",
       " 'women',\n",
       " 'significantli',\n",
       " 'better',\n",
       " 'prognosi',\n",
       " 'compar',\n",
       " 'men',\n",
       " 'number',\n",
       " 'genet',\n",
       " 'alter',\n",
       " 'occur',\n",
       " 'lung',\n",
       " 'cancer',\n",
       " 'exampl',\n",
       " 'nsclc',\n",
       " 'mutat',\n",
       " 'kra',\n",
       " 'p53',\n",
       " 'egfr',\n",
       " 'met',\n",
       " 'identifi',\n",
       " 'mani',\n",
       " 'pathway',\n",
       " 'especi',\n",
       " 'receptor',\n",
       " 'tyrosin',\n",
       " 'kinas',\n",
       " 'rtk',\n",
       " 'control',\n",
       " 'c-cbl',\n",
       " 'cbl',\n",
       " 'casita',\n",
       " 'b-lineag',\n",
       " 'lymphoma',\n",
       " 'mammalian',\n",
       " 'gene',\n",
       " 'locat',\n",
       " 'human',\n",
       " 'chromosom',\n",
       " '11q23',\n",
       " 'involv',\n",
       " 'cell',\n",
       " 'signal',\n",
       " 'protein',\n",
       " 'ubiquitin',\n",
       " 'cbl',\n",
       " 'protein',\n",
       " 'belong',\n",
       " 'ring',\n",
       " 'finger',\n",
       " 'class',\n",
       " 'ubiquitin',\n",
       " 'ligas',\n",
       " 'homologu',\n",
       " 'c-cbl',\n",
       " 'cbl-b',\n",
       " 'cbl-3',\n",
       " 'c-cbl',\n",
       " 'cbl-b',\n",
       " 'gene',\n",
       " 'ubiquit',\n",
       " 'express',\n",
       " 'highest',\n",
       " 'level',\n",
       " 'hematopoiet',\n",
       " 'tissu',\n",
       " 'c-cbl',\n",
       " 'consist',\n",
       " 'region',\n",
       " 'encod',\n",
       " 'function',\n",
       " 'distinct',\n",
       " 'protein',\n",
       " 'domain',\n",
       " 'n-termin',\n",
       " 'tyrosin',\n",
       " 'kinas',\n",
       " 'bind',\n",
       " 'tkb',\n",
       " 'domain',\n",
       " 'linker',\n",
       " 'region',\n",
       " 'catalyt',\n",
       " 'ring',\n",
       " 'finger',\n",
       " 'domain',\n",
       " 'proline-rich',\n",
       " 'region',\n",
       " 'c-termin',\n",
       " 'ubiquitin-associ',\n",
       " 'uba',\n",
       " 'domain',\n",
       " 'also',\n",
       " 'overlap',\n",
       " 'leucine-zipp',\n",
       " 'domain',\n",
       " 'tkb',\n",
       " 'ring',\n",
       " 'finger',\n",
       " 'domain',\n",
       " 'essenti',\n",
       " 'ligand-induc',\n",
       " 'ubiquitin',\n",
       " 'rtk',\n",
       " 'ring',\n",
       " 'finger',\n",
       " 'domain',\n",
       " 'requir',\n",
       " 'recruit',\n",
       " 'ubiquitin-conjug',\n",
       " 'enzym',\n",
       " 'tkb',\n",
       " 'domain',\n",
       " 'includ',\n",
       " 'four-helix',\n",
       " 'bundl',\n",
       " 'calcium-bid',\n",
       " 'hand',\n",
       " 'modifi',\n",
       " 'sh2',\n",
       " 'domain',\n",
       " 'bind',\n",
       " 'phosphotyrosin',\n",
       " 'residu',\n",
       " 'addit',\n",
       " 'proline-rich',\n",
       " 'region',\n",
       " 'c-cbl',\n",
       " 'associ',\n",
       " 'sh3',\n",
       " 'domain',\n",
       " 'grb2',\n",
       " 'indirectli',\n",
       " 'recruit',\n",
       " 'c-cbl',\n",
       " 'rtk',\n",
       " 'via',\n",
       " 'grb2',\n",
       " 'adaptor',\n",
       " 'protein',\n",
       " 'c-cbl',\n",
       " 'also',\n",
       " 'bind',\n",
       " 'egfr',\n",
       " 'act',\n",
       " 'target',\n",
       " 'egfr',\n",
       " 'ubiquitin',\n",
       " 'degrad',\n",
       " 'furthermor',\n",
       " 'cbl',\n",
       " 'desensit',\n",
       " 'egf',\n",
       " 'signal',\n",
       " 'oppos',\n",
       " 'cellular',\n",
       " 'prolifer',\n",
       " 'induc',\n",
       " 'egf',\n",
       " 'egf',\n",
       " 'activ',\n",
       " 'also',\n",
       " 'appear',\n",
       " 'activ',\n",
       " 'tyrosin',\n",
       " 'kinas',\n",
       " 'src',\n",
       " 'phosphoryl',\n",
       " 'c-cbl',\n",
       " 'turn',\n",
       " 'activ',\n",
       " 'ubiquitin',\n",
       " 'degrad',\n",
       " 'egfr',\n",
       " 'recent',\n",
       " 'studi',\n",
       " 'show',\n",
       " 'defect',\n",
       " 'endocytosi',\n",
       " 'egfr',\n",
       " 'character',\n",
       " 'delet',\n",
       " 'mutant',\n",
       " 'point',\n",
       " 'mutat',\n",
       " 'l858r',\n",
       " 'wherebi',\n",
       " 'associ',\n",
       " 'c-cbl',\n",
       " 'subsequ',\n",
       " 'ubiquitin',\n",
       " 'impair',\n",
       " 'recent',\n",
       " 'first',\n",
       " 'human',\n",
       " 'c-cbl',\n",
       " 'mutat',\n",
       " 'report',\n",
       " 'acut',\n",
       " 'myeloid',\n",
       " 'leukemia',\n",
       " 'aml',\n",
       " 'patient',\n",
       " 'mutat',\n",
       " 'r420q',\n",
       " 'inhibit',\n",
       " 'fms-like',\n",
       " 'tyrosin',\n",
       " 'kinas',\n",
       " 'flt3',\n",
       " 'intern',\n",
       " 'ubiquitin',\n",
       " 'activ',\n",
       " 'import',\n",
       " 'oncogenesi',\n",
       " 'c-cbl',\n",
       " 'dual',\n",
       " 'separ',\n",
       " 'function',\n",
       " 'signal',\n",
       " 'transduct',\n",
       " 'molecul',\n",
       " 'previous',\n",
       " 'shown',\n",
       " 'c-cbl',\n",
       " 'import',\n",
       " 'bind',\n",
       " 'crkl',\n",
       " 'bcr',\n",
       " 'abl',\n",
       " 'hematopoiet',\n",
       " 'cell',\n",
       " 'also',\n",
       " 'bind',\n",
       " 'modul',\n",
       " 'function',\n",
       " 'cytoskeleton',\n",
       " 'bind',\n",
       " 'protein',\n",
       " 'like',\n",
       " 'talin',\n",
       " 'paxillin',\n",
       " 'tkb',\n",
       " 'domain',\n",
       " 'import',\n",
       " 'bind',\n",
       " 'number',\n",
       " 'molecul',\n",
       " 'function',\n",
       " 'signal',\n",
       " 'transduct',\n",
       " 'given',\n",
       " 'critic',\n",
       " 'role',\n",
       " 'cbl',\n",
       " 'normal',\n",
       " 'homeostasi',\n",
       " 'cancer',\n",
       " 'hypothes',\n",
       " 'might',\n",
       " 'mutat',\n",
       " 'lung',\n",
       " 'cancer',\n",
       " 'studi',\n",
       " 'report',\n",
       " 'novel',\n",
       " 'c-cbl',\n",
       " 'somat',\n",
       " 'mutat',\n",
       " 's80n',\n",
       " 'h94i',\n",
       " 'q249e',\n",
       " 'w802',\n",
       " 'caucasian',\n",
       " 'taiwanes',\n",
       " 'african-american',\n",
       " 'lung',\n",
       " 'cancer',\n",
       " 'patient',\n",
       " 'respect',\n",
       " 'express',\n",
       " 'mutat',\n",
       " 'nsclc',\n",
       " 'cell',\n",
       " 'line',\n",
       " 'lead',\n",
       " 'increas',\n",
       " 'prolifer',\n",
       " 'cell',\n",
       " 'motil',\n",
       " 'show',\n",
       " 'c-cbl',\n",
       " 'mutat',\n",
       " 'occur',\n",
       " 'without',\n",
       " 'met',\n",
       " 'egfr',\n",
       " 'mutat',\n",
       " 'mutual',\n",
       " 'exclus',\n",
       " 'loh',\n",
       " 'c-cbl',\n",
       " 'locu',\n",
       " 'addit',\n",
       " 'c-cbl',\n",
       " 'loh',\n",
       " 'associ',\n",
       " 'either',\n",
       " 'met',\n",
       " 'egfr',\n",
       " 'mutat',\n",
       " 'thu',\n",
       " 'hypothes',\n",
       " 'c-cbl',\n",
       " 'mutat',\n",
       " 'might',\n",
       " 'contribut',\n",
       " 'oncogen',\n",
       " 'potenti',\n",
       " 'met',\n",
       " 'egfr',\n",
       " 'lung',\n",
       " 'cancer',\n",
       " 'method',\n",
       " 'ethic',\n",
       " 'statement',\n",
       " 'written',\n",
       " 'consent',\n",
       " 'research',\n",
       " 'human',\n",
       " 'subject',\n",
       " 'obtain',\n",
       " 'institut',\n",
       " 'review',\n",
       " 'board',\n",
       " 'univers',\n",
       " 'chicago',\n",
       " 'cover',\n",
       " 'research',\n",
       " 'perform',\n",
       " 'laboratori',\n",
       " 'follow',\n",
       " 'contact',\n",
       " 'inform',\n",
       " 'institut',\n",
       " 'review',\n",
       " 'board',\n",
       " 'univers',\n",
       " 'chicago',\n",
       " 'mcgiffert',\n",
       " 'hall',\n",
       " '5751',\n",
       " 'woodlawn',\n",
       " 'ave',\n",
       " '2nd',\n",
       " 'floor',\n",
       " 'chicago',\n",
       " '60637',\n",
       " 'written',\n",
       " 'inform',\n",
       " 'consent',\n",
       " 'receiv',\n",
       " 'patient',\n",
       " 'whose',\n",
       " 'tissu',\n",
       " 'sampl',\n",
       " 'use',\n",
       " 'studi',\n",
       " 'tissu',\n",
       " 'sampl',\n",
       " 'lung',\n",
       " 'cancer',\n",
       " 'tissu',\n",
       " 'pair',\n",
       " 'adjac',\n",
       " 'normal',\n",
       " 'lung',\n",
       " 'tissu',\n",
       " 'obtain',\n",
       " 'caucasian',\n",
       " 'african-american',\n",
       " 'taiwanes',\n",
       " 'nsclc',\n",
       " 'patient',\n",
       " 'recruit',\n",
       " 'univers',\n",
       " 'chicago',\n",
       " 'hospit',\n",
       " 'chicago',\n",
       " 'usa',\n",
       " 'caucasian',\n",
       " 'african-american',\n",
       " 'patient',\n",
       " 'taipei',\n",
       " 'veteran',\n",
       " 'gener',\n",
       " 'hospit',\n",
       " 'taiwan',\n",
       " 'taiwanes',\n",
       " 'patient',\n",
       " 'obtain',\n",
       " 'appropri',\n",
       " 'institut',\n",
       " 'review',\n",
       " 'board',\n",
       " 'permiss',\n",
       " 'inform',\n",
       " 'consent',\n",
       " 'patient',\n",
       " '119',\n",
       " 'sampl',\n",
       " 'men',\n",
       " 'women',\n",
       " 'unknown',\n",
       " 'age',\n",
       " 'diagnosi',\n",
       " 'rang',\n",
       " 'year',\n",
       " 'term',\n",
       " 'tumor',\n",
       " 'type',\n",
       " 'adenocarcinoma',\n",
       " 'squamou',\n",
       " 'cell',\n",
       " 'carcinoma',\n",
       " 'larg',\n",
       " 'cell',\n",
       " 'carcinoma',\n",
       " 'stage',\n",
       " 'stage',\n",
       " 'stage',\n",
       " 'iii',\n",
       " 'stage',\n",
       " 'tabl',\n",
       " 'cell',\n",
       " 'cultur',\n",
       " 'human',\n",
       " 'non-smal',\n",
       " 'cell',\n",
       " 'lung',\n",
       " 'carcinoma',\n",
       " 'cell',\n",
       " 'a549',\n",
       " 'h358',\n",
       " 'maintain',\n",
       " 'dmem',\n",
       " 'rpmi-1640',\n",
       " 'respect',\n",
       " 'human',\n",
       " 'embryon',\n",
       " 'kidney',\n",
       " '293t',\n",
       " 'cell',\n",
       " 'cultur',\n",
       " 'dmem',\n",
       " 'media',\n",
       " 'supplement',\n",
       " 'fetal',\n",
       " 'bovin',\n",
       " 'serum',\n",
       " '100',\n",
       " 'unit',\n",
       " 'penicillin',\n",
       " '100',\n",
       " 'streptomycin',\n",
       " 'invitrogen',\n",
       " 'carlsbad',\n",
       " 'cell',\n",
       " 'cultur',\n",
       " 'humidifi',\n",
       " 'incub',\n",
       " 'contain',\n",
       " 'co2',\n",
       " 'c-cbl',\n",
       " 'gene',\n",
       " 'mutat',\n",
       " 'analysi',\n",
       " 'exon',\n",
       " 'c-cbl',\n",
       " 'gene',\n",
       " 'individu',\n",
       " 'amplifi',\n",
       " 'polymeras',\n",
       " 'chain',\n",
       " 'reaction',\n",
       " 'pcr',\n",
       " 'primer',\n",
       " 'list',\n",
       " 'tabl',\n",
       " 'pcr',\n",
       " 'condit',\n",
       " 'cycl',\n",
       " 'minut',\n",
       " 'cycl',\n",
       " 'second',\n",
       " 'second',\n",
       " 'minut',\n",
       " 'cycl',\n",
       " 'minut',\n",
       " 'pcr',\n",
       " 'product',\n",
       " 'treat',\n",
       " 'exosap-it',\n",
       " 'usb',\n",
       " 'corpor',\n",
       " 'cleveland',\n",
       " 'sequenc',\n",
       " 'big-dy',\n",
       " 'termin',\n",
       " 'chemistri',\n",
       " 'appli',\n",
       " 'biosystem',\n",
       " 'foster',\n",
       " 'citi',\n",
       " 'sequenc',\n",
       " 'perform',\n",
       " 'forward',\n",
       " 'code',\n",
       " 'strand',\n",
       " 'confirm',\n",
       " 'c-cbl',\n",
       " 'alter',\n",
       " 'perform',\n",
       " 'sequenc',\n",
       " 'revers',\n",
       " 'strand',\n",
       " 'well',\n",
       " 'chromatogram',\n",
       " 'analyz',\n",
       " 'mutat',\n",
       " 'use',\n",
       " 'mutat',\n",
       " 'surveyor',\n",
       " 'softgenet',\n",
       " 'state',\n",
       " 'colleg',\n",
       " 'plasmid',\n",
       " 'construct',\n",
       " 'site-direct',\n",
       " 'mutagenesi',\n",
       " 'wild-typ',\n",
       " 'c-cbl',\n",
       " 'cdna',\n",
       " 'insert',\n",
       " 'subclon',\n",
       " 'paltermax',\n",
       " 'express',\n",
       " 'vector',\n",
       " 'use',\n",
       " 'xhoi',\n",
       " 'sali',\n",
       " 'restrict',\n",
       " 'enzym',\n",
       " 'site',\n",
       " 'promega',\n",
       " 'madison',\n",
       " 'use',\n",
       " 'parent',\n",
       " 'plasmid',\n",
       " 'paltermax-c-cbl',\n",
       " 'tkb',\n",
       " 'domain',\n",
       " 'doubl',\n",
       " 'mutat',\n",
       " 's80n',\n",
       " 'h94i',\n",
       " 'point',\n",
       " 'mutat',\n",
       " 'q249e',\n",
       " 'c-termin',\n",
       " 'point',\n",
       " 'mutat',\n",
       " 'w802',\n",
       " 'c-cbl',\n",
       " 'creat',\n",
       " 'use',\n",
       " 'follow',\n",
       " 'primer',\n",
       " '-gctggcgctaaagaataacccaccttatatcttagac-3',\n",
       " '-ctaccagatacctaccagtatctccgtactatcttgtc-3',\n",
       " 'doubl',\n",
       " 'mutat',\n",
       " 's80n',\n",
       " 'h94i',\n",
       " '-ctttacccgactctttgagccctggtcctctttgc-3',\n",
       " 'q249e',\n",
       " '-cagctcctcctttggctgattgtctctggatggtgatc-3',\n",
       " 'w802',\n",
       " 'along',\n",
       " 'complementari',\n",
       " 'primer',\n",
       " 'use',\n",
       " 'quickchang',\n",
       " 'site-direct',\n",
       " 'mutagenesi',\n",
       " 'kit',\n",
       " 'stratagen',\n",
       " 'jolla',\n",
       " 'accord',\n",
       " 'manufactur',\n",
       " 'instruct',\n",
       " 'construct',\n",
       " 'confirm',\n",
       " 'point',\n",
       " 'mutat',\n",
       " 'standard',\n",
       " 'dna',\n",
       " 'sequenc',\n",
       " 'strand',\n",
       " 'loss',\n",
       " 'heterozygos',\n",
       " 'loh',\n",
       " 'analysi',\n",
       " 'microsatellit',\n",
       " 'chromosom',\n",
       " '11q',\n",
       " 'within',\n",
       " '200',\n",
       " 'downstream',\n",
       " 'c-cbl',\n",
       " 'gene',\n",
       " 'control',\n",
       " 'marker',\n",
       " '11p',\n",
       " 'select',\n",
       " 'analysi',\n",
       " 'tabl',\n",
       " 'establish',\n",
       " 'microsatellit',\n",
       " 'marker',\n",
       " 'respect',\n",
       " 'primer',\n",
       " 'sequenc',\n",
       " 'select',\n",
       " 'geneloc',\n",
       " 'databas',\n",
       " 'http',\n",
       " 'genecard',\n",
       " 'weizmann',\n",
       " 'geneloc',\n",
       " 'index',\n",
       " 'shtml',\n",
       " 'weizmann',\n",
       " 'institut',\n",
       " 'scienc',\n",
       " 'rehovot',\n",
       " 'israel',\n",
       " 'primer',\n",
       " 'custom',\n",
       " 'design',\n",
       " 'forward',\n",
       " 'primer',\n",
       " 'fluoresc',\n",
       " 'label',\n",
       " 'end',\n",
       " 'fam',\n",
       " 'pet',\n",
       " 'ned',\n",
       " 'vic',\n",
       " 'appli',\n",
       " 'biosystem',\n",
       " 'primer',\n",
       " 'anneal',\n",
       " 'temperatur',\n",
       " 'duplex',\n",
       " 'score',\n",
       " 'evalu',\n",
       " 'nist',\n",
       " 'primer',\n",
       " 'tool',\n",
       " 'http',\n",
       " 'yellow',\n",
       " 'nist',\n",
       " 'gov',\n",
       " '8444',\n",
       " 'dnaanalysi',\n",
       " 'primertoolspag',\n",
       " 'nation',\n",
       " 'institut',\n",
       " 'standard',\n",
       " 'technolog',\n",
       " 'gaithersburg',\n",
       " 'primer',\n",
       " 'verifi',\n",
       " 'perform',\n",
       " 'pcr',\n",
       " 'control',\n",
       " 'dna',\n",
       " 'isol',\n",
       " 'tk6',\n",
       " 'cell',\n",
       " 'resolv',\n",
       " 'product',\n",
       " 'agaros',\n",
       " 'gel',\n",
       " 'band',\n",
       " 'visual',\n",
       " 'transillumin',\n",
       " 'genom',\n",
       " 'dna',\n",
       " 'extract',\n",
       " 'tumor',\n",
       " 'sampl',\n",
       " 'pair',\n",
       " 'normal',\n",
       " 'lung',\n",
       " 'tissu',\n",
       " 'primer',\n",
       " 'group',\n",
       " 'multiplex',\n",
       " 'combin',\n",
       " 'shown',\n",
       " 'tabl',\n",
       " 'marker',\n",
       " 'd11s929',\n",
       " 'serv',\n",
       " 'intern',\n",
       " 'control',\n",
       " 'check',\n",
       " 'consist',\n",
       " 'pcr',\n",
       " 'peak',\n",
       " 'capillari',\n",
       " 'electrophoresi',\n",
       " 'multiplex',\n",
       " 'pcr',\n",
       " 'carri',\n",
       " 'volum',\n",
       " 'contain',\n",
       " 'genom',\n",
       " 'dna',\n",
       " 'primer',\n",
       " 'total',\n",
       " 'primer',\n",
       " 'pair',\n",
       " '400',\n",
       " 'dntp',\n",
       " 'pcr',\n",
       " 'buffer',\n",
       " 'contain',\n",
       " 'mgcl2',\n",
       " 'taq',\n",
       " 'dna',\n",
       " 'polymeras',\n",
       " 'pcr',\n",
       " 'perform',\n",
       " 'abi',\n",
       " 'geneamp',\n",
       " '9700',\n",
       " 'pcr',\n",
       " 'system',\n",
       " 'follow',\n",
       " 'condit',\n",
       " 'min',\n",
       " 'cycl',\n",
       " 'sec',\n",
       " 'min',\n",
       " 'min',\n",
       " 'min',\n",
       " 'pcr',\n",
       " 'product',\n",
       " 'separ',\n",
       " 'capillari',\n",
       " 'electrophoresi',\n",
       " 'abi',\n",
       " '3130xl',\n",
       " 'dna',\n",
       " 'analyz',\n",
       " 'chromatogram',\n",
       " 'analyz',\n",
       " 'peak',\n",
       " 'scanner',\n",
       " 'genemapp',\n",
       " 'softwar',\n",
       " 'appli',\n",
       " 'biosystem',\n",
       " 'allel',\n",
       " 'alter',\n",
       " 'area',\n",
       " 'peak',\n",
       " 'produc',\n",
       " 'dna',\n",
       " 'pcr',\n",
       " 'product',\n",
       " 'quantifi',\n",
       " 'allel',\n",
       " 'ratio',\n",
       " 'allel',\n",
       " 'area',\n",
       " 'calcul',\n",
       " 'tumor',\n",
       " 'pair',\n",
       " 'normal',\n",
       " 'dna',\n",
       " 'sampl',\n",
       " 'qloh',\n",
       " 'allel',\n",
       " 'ratio',\n",
       " 'tumor',\n",
       " 'peak',\n",
       " 'divid',\n",
       " 'allel',\n",
       " 'ratio',\n",
       " 'pair',\n",
       " 'normal',\n",
       " 'sampl',\n",
       " 'c-cbl',\n",
       " 'least',\n",
       " '11q',\n",
       " 'marker',\n",
       " ...]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "X.processed[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'X_train' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-7-897aa868a671>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mxgboost\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mX_train\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mX_test\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0my_train\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0my_test\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'X_train' is not defined"
     ]
    }
   ],
   "source": [
    "import xgboost\n",
    "X_train,X_test,y_train,y_test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.97534338129789677"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn.model_selection import train_test_split\n",
    "def return_same(x):\n",
    "    return x\n",
    "from sklearn.pipeline import Pipeline\n",
    "from sklearn.feature_extraction.text import HashingVectorizer\n",
    "from sklearn.metrics import log_loss\n",
    "x_train,x_test,y_train,y_test = train_test_split(X.processed, X.Class)\n",
    "pipe = Pipeline([('encoder',HashingVectorizer(analyzer=return_same)),\n",
    "                 ('lr',xgboost.XGBClassifier(n_jobs=-1))])\n",
    "pipe.fit(x_train,y_train)\n",
    "y_pred = pipe.predict_proba(x_test)\n",
    "log_loss(y_test,y_pred)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1.0325121435994635"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn.feature_extraction.text import TfidfVectorizer\n",
    "x_train,x_test,y_train,y_test = train_test_split(X.processed, X.Class)\n",
    "pipe = Pipeline([('encoder',TfidfVectorizer(analyzer=return_same)),\n",
    "                 ('lr',xgboost.XGBClassifier(n_jobs=-1))])\n",
    "pipe.fit(x_train,y_train)\n",
    "y_pred = pipe.predict_proba(x_test)\n",
    "log_loss(y_test,y_pred)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'encoder': TfidfVectorizer(analyzer=<function return_same at 0x7f8df0e49f28>,\n",
       "         binary=False, decode_error='strict', dtype=<class 'numpy.int64'>,\n",
       "         encoding='utf-8', input='content', lowercase=True, max_df=1.0,\n",
       "         max_features=None, min_df=1, ngram_range=(1, 1), norm='l2',\n",
       "         preprocessor=None, smooth_idf=True, stop_words=None,\n",
       "         strip_accents=None, sublinear_tf=False,\n",
       "         token_pattern='(?u)\\\\b\\\\w\\\\w+\\\\b', tokenizer=None, use_idf=True,\n",
       "         vocabulary=None),\n",
       " 'encoder__analyzer': <function __main__.return_same>,\n",
       " 'encoder__binary': False,\n",
       " 'encoder__decode_error': 'strict',\n",
       " 'encoder__dtype': numpy.int64,\n",
       " 'encoder__encoding': 'utf-8',\n",
       " 'encoder__input': 'content',\n",
       " 'encoder__lowercase': True,\n",
       " 'encoder__max_df': 1.0,\n",
       " 'encoder__max_features': None,\n",
       " 'encoder__min_df': 1,\n",
       " 'encoder__ngram_range': (1, 1),\n",
       " 'encoder__norm': 'l2',\n",
       " 'encoder__preprocessor': None,\n",
       " 'encoder__smooth_idf': True,\n",
       " 'encoder__stop_words': None,\n",
       " 'encoder__strip_accents': None,\n",
       " 'encoder__sublinear_tf': False,\n",
       " 'encoder__token_pattern': '(?u)\\\\b\\\\w\\\\w+\\\\b',\n",
       " 'encoder__tokenizer': None,\n",
       " 'encoder__use_idf': True,\n",
       " 'encoder__vocabulary': None,\n",
       " 'lr': XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,\n",
       "        colsample_bytree=1, gamma=0, learning_rate=0.1, max_delta_step=0,\n",
       "        max_depth=3, min_child_weight=1, missing=None, n_estimators=100,\n",
       "        n_jobs=-1, nthread=None, objective='multi:softprob', random_state=0,\n",
       "        reg_alpha=0, reg_lambda=1, scale_pos_weight=1, seed=None,\n",
       "        silent=True, subsample=1),\n",
       " 'lr__base_score': 0.5,\n",
       " 'lr__booster': 'gbtree',\n",
       " 'lr__colsample_bylevel': 1,\n",
       " 'lr__colsample_bytree': 1,\n",
       " 'lr__gamma': 0,\n",
       " 'lr__learning_rate': 0.1,\n",
       " 'lr__max_delta_step': 0,\n",
       " 'lr__max_depth': 3,\n",
       " 'lr__min_child_weight': 1,\n",
       " 'lr__missing': None,\n",
       " 'lr__n_estimators': 100,\n",
       " 'lr__n_jobs': -1,\n",
       " 'lr__nthread': None,\n",
       " 'lr__objective': 'multi:softprob',\n",
       " 'lr__random_state': 0,\n",
       " 'lr__reg_alpha': 0,\n",
       " 'lr__reg_lambda': 1,\n",
       " 'lr__scale_pos_weight': 1,\n",
       " 'lr__seed': None,\n",
       " 'lr__silent': True,\n",
       " 'lr__subsample': 1,\n",
       " 'steps': [('encoder',\n",
       "   TfidfVectorizer(analyzer=<function return_same at 0x7f8df0e49f28>,\n",
       "           binary=False, decode_error='strict', dtype=<class 'numpy.int64'>,\n",
       "           encoding='utf-8', input='content', lowercase=True, max_df=1.0,\n",
       "           max_features=None, min_df=1, ngram_range=(1, 1), norm='l2',\n",
       "           preprocessor=None, smooth_idf=True, stop_words=None,\n",
       "           strip_accents=None, sublinear_tf=False,\n",
       "           token_pattern='(?u)\\\\b\\\\w\\\\w+\\\\b', tokenizer=None, use_idf=True,\n",
       "           vocabulary=None)),\n",
       "  ('lr', XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,\n",
       "          colsample_bytree=1, gamma=0, learning_rate=0.1, max_delta_step=0,\n",
       "          max_depth=3, min_child_weight=1, missing=None, n_estimators=100,\n",
       "          n_jobs=-1, nthread=None, objective='multi:softprob', random_state=0,\n",
       "          reg_alpha=0, reg_lambda=1, scale_pos_weight=1, seed=None,\n",
       "          silent=True, subsample=1))]}"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pipe.get_params()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GridSearchCV(cv=None, error_score='raise',\n",
       "       estimator=Pipeline(steps=[('encoder', TfidfVectorizer(analyzer=<function return_same at 0x7ff93f677e18>,\n",
       "        binary=False, decode_error='strict', dtype=<class 'numpy.int64'>,\n",
       "        encoding='utf-8', input='content', lowercase=True, max_df=1.0,\n",
       "        max_features=None, min_df=1, ngram_range=(1, 1), norm='l2',\n",
       " ...tate=0, reg_alpha=0, reg_lambda=1, scale_pos_weight=1,\n",
       "       seed=None, silent=True, subsample=1))]),\n",
       "       fit_params={}, iid=True, n_jobs=1,\n",
       "       param_grid={'encoder__min_df': [0.01, 0.05, 0.1], 'encoder__max_df': [0.9, 0.7, 0.5]},\n",
       "       pre_dispatch='2*n_jobs', refit=True, return_train_score=True,\n",
       "       scoring='neg_log_loss', verbose=0)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn.model_selection import GridSearchCV\n",
    "from sklearn.model_selection import train_test_split\n",
    "def return_same(x):\n",
    "    return x\n",
    "from sklearn.pipeline import Pipeline\n",
    "from sklearn.feature_extraction.text import HashingVectorizer\n",
    "from sklearn.feature_extraction.text import TfidfVectorizer\n",
    "from sklearn.metrics import log_loss\n",
    "pipe = Pipeline([('encoder',TfidfVectorizer(analyzer=return_same)),\n",
    "                 ('lr',xgboost.XGBClassifier(n_jobs=-1))])\n",
    "param_grid = {'encoder__min_df':[.01,.05,.1],'encoder__max_df':[.9,.7,.5]}\n",
    "gs = GridSearchCV(pipe, param_grid, scoring = 'neg_log_loss',n_jobs=1)\n",
    "gs.fit(X.processed,X.Class)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "-1.7098232543612506"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gs.best_score_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/ec2-user/yes/lib/python3.6/site-packages/sklearn/model_selection/_search.py:667: DeprecationWarning: The grid_scores_ attribute was deprecated in version 0.18 in favor of the more elaborate cv_results_ attribute. The grid_scores_ attribute will not be available from 0.20\n",
      "  DeprecationWarning)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[mean: -1.72409, std: 0.12176, params: {'encoder__max_df': 0.9, 'encoder__min_df': 0.01},\n",
       " mean: -1.70982, std: 0.10712, params: {'encoder__max_df': 0.9, 'encoder__min_df': 0.05},\n",
       " mean: -1.75826, std: 0.07061, params: {'encoder__max_df': 0.9, 'encoder__min_df': 0.1},\n",
       " mean: -1.72592, std: 0.09993, params: {'encoder__max_df': 0.7, 'encoder__min_df': 0.01},\n",
       " mean: -1.73647, std: 0.09111, params: {'encoder__max_df': 0.7, 'encoder__min_df': 0.05},\n",
       " mean: -1.74383, std: 0.07543, params: {'encoder__max_df': 0.7, 'encoder__min_df': 0.1},\n",
       " mean: -1.76090, std: 0.04831, params: {'encoder__max_df': 0.5, 'encoder__min_df': 0.01},\n",
       " mean: -1.76456, std: 0.06481, params: {'encoder__max_df': 0.5, 'encoder__min_df': 0.05},\n",
       " mean: -1.77976, std: 0.03901, params: {'encoder__max_df': 0.5, 'encoder__min_df': 0.1}]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gs.grid_scores_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'encoder': HashingVectorizer(analyzer=<function return_same at 0x7ff93f677e18>,\n",
       "          binary=False, decode_error='strict',\n",
       "          dtype=<class 'numpy.float64'>, encoding='utf-8', input='content',\n",
       "          lowercase=True, n_features=1048576, ngram_range=(1, 1),\n",
       "          non_negative=False, norm='l2', preprocessor=None, stop_words=None,\n",
       "          strip_accents=None, token_pattern='(?u)\\\\b\\\\w\\\\w+\\\\b',\n",
       "          tokenizer=None),\n",
       " 'encoder__analyzer': <function __main__.return_same>,\n",
       " 'encoder__binary': False,\n",
       " 'encoder__decode_error': 'strict',\n",
       " 'encoder__dtype': numpy.float64,\n",
       " 'encoder__encoding': 'utf-8',\n",
       " 'encoder__input': 'content',\n",
       " 'encoder__lowercase': True,\n",
       " 'encoder__n_features': 1048576,\n",
       " 'encoder__ngram_range': (1, 1),\n",
       " 'encoder__non_negative': False,\n",
       " 'encoder__norm': 'l2',\n",
       " 'encoder__preprocessor': None,\n",
       " 'encoder__stop_words': None,\n",
       " 'encoder__strip_accents': None,\n",
       " 'encoder__token_pattern': '(?u)\\\\b\\\\w\\\\w+\\\\b',\n",
       " 'encoder__tokenizer': None,\n",
       " 'lr': XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,\n",
       "        colsample_bytree=1, gamma=0, learning_rate=0.1, max_delta_step=0,\n",
       "        max_depth=3, min_child_weight=1, missing=None, n_estimators=100,\n",
       "        n_jobs=-1, nthread=None, objective='binary:logistic',\n",
       "        random_state=0, reg_alpha=0, reg_lambda=1, scale_pos_weight=1,\n",
       "        seed=None, silent=True, subsample=1),\n",
       " 'lr__base_score': 0.5,\n",
       " 'lr__booster': 'gbtree',\n",
       " 'lr__colsample_bylevel': 1,\n",
       " 'lr__colsample_bytree': 1,\n",
       " 'lr__gamma': 0,\n",
       " 'lr__learning_rate': 0.1,\n",
       " 'lr__max_delta_step': 0,\n",
       " 'lr__max_depth': 3,\n",
       " 'lr__min_child_weight': 1,\n",
       " 'lr__missing': None,\n",
       " 'lr__n_estimators': 100,\n",
       " 'lr__n_jobs': -1,\n",
       " 'lr__nthread': None,\n",
       " 'lr__objective': 'binary:logistic',\n",
       " 'lr__random_state': 0,\n",
       " 'lr__reg_alpha': 0,\n",
       " 'lr__reg_lambda': 1,\n",
       " 'lr__scale_pos_weight': 1,\n",
       " 'lr__seed': None,\n",
       " 'lr__silent': True,\n",
       " 'lr__subsample': 1,\n",
       " 'steps': [('encoder',\n",
       "   HashingVectorizer(analyzer=<function return_same at 0x7ff93f677e18>,\n",
       "            binary=False, decode_error='strict',\n",
       "            dtype=<class 'numpy.float64'>, encoding='utf-8', input='content',\n",
       "            lowercase=True, n_features=1048576, ngram_range=(1, 1),\n",
       "            non_negative=False, norm='l2', preprocessor=None, stop_words=None,\n",
       "            strip_accents=None, token_pattern='(?u)\\\\b\\\\w\\\\w+\\\\b',\n",
       "            tokenizer=None)),\n",
       "  ('lr', XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,\n",
       "          colsample_bytree=1, gamma=0, learning_rate=0.1, max_delta_step=0,\n",
       "          max_depth=3, min_child_weight=1, missing=None, n_estimators=100,\n",
       "          n_jobs=-1, nthread=None, objective='binary:logistic',\n",
       "          random_state=0, reg_alpha=0, reg_lambda=1, scale_pos_weight=1,\n",
       "          seed=None, silent=True, subsample=1))]}"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x_train,x_test,y_train,y_test = train_test_split(X.processed, X.Class)\n",
    "pipe = Pipeline([('encoder',HashingVectorizer(analyzer=return_same)),\n",
    "                 ('lr',xgboost.XGBClassifier(n_jobs=-1))])\n",
    "pipe.get_params()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-1.45517661443\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/ec2-user/yes/lib/python3.6/site-packages/sklearn/model_selection/_search.py:667: DeprecationWarning: The grid_scores_ attribute was deprecated in version 0.18 in favor of the more elaborate cv_results_ attribute. The grid_scores_ attribute will not be available from 0.20\n",
      "  DeprecationWarning)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[mean: -1.45518, std: 0.14744, params: {'lr__max_depth': 2},\n",
       " mean: -1.51341, std: 0.17457, params: {'lr__max_depth': 3},\n",
       " mean: -1.58491, std: 0.20233, params: {'lr__max_depth': 4}]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn.model_selection import GridSearchCV\n",
    "from sklearn.model_selection import train_test_split\n",
    "def return_same(x):\n",
    "    return x\n",
    "from sklearn.pipeline import Pipeline\n",
    "from sklearn.feature_extraction.text import HashingVectorizer\n",
    "from sklearn.feature_extraction.text import TfidfVectorizer\n",
    "from sklearn.metrics import log_loss\n",
    "pipe = Pipeline([('encoder',HashingVectorizer(analyzer=return_same)),\n",
    "                 ('lr',xgboost.XGBClassifier(n_jobs=-1))])\n",
    "param_grid = {'lr__max_depth':[2,3,4]}\n",
    "gs = GridSearchCV(pipe, param_grid, scoring = 'neg_log_loss',n_jobs=1,cv=5)\n",
    "gs.fit(X.processed,X.Class)\n",
    "print(gs.best_score_)\n",
    "gs.grid_scores_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-1.49456980637\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/ec2-user/yes/lib/python3.6/site-packages/sklearn/model_selection/_search.py:667: DeprecationWarning: The grid_scores_ attribute was deprecated in version 0.18 in favor of the more elaborate cv_results_ attribute. The grid_scores_ attribute will not be available from 0.20\n",
      "  DeprecationWarning)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[mean: -1.49457, std: 0.09700, params: {'lr__max_depth': 2}]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pipe = Pipeline([('encoder',HashingVectorizer(analyzer=return_same)),\n",
    "                 ('lr',xgboost.XGBClassifier(n_jobs=-1,learning_rate=.01,n_estimators=300))])\n",
    "param_grid = {'lr__max_depth':[2]}\n",
    "gs = GridSearchCV(pipe, param_grid, scoring = 'neg_log_loss',n_jobs=1,cv=5)\n",
    "gs.fit(X.processed,X.Class)\n",
    "print(gs.best_score_)\n",
    "gs.grid_scores_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'mean_fit_time': array([ 548.13313084]),\n",
       " 'mean_score_time': array([ 3.06792235]),\n",
       " 'mean_test_score': array([-1.49456981]),\n",
       " 'mean_train_score': array([-1.08604161]),\n",
       " 'param_lr__max_depth': masked_array(data = [2],\n",
       "              mask = [False],\n",
       "        fill_value = ?),\n",
       " 'params': ({'lr__max_depth': 2},),\n",
       " 'rank_test_score': array([1], dtype=int32),\n",
       " 'split0_test_score': array([-1.55355988]),\n",
       " 'split0_train_score': array([-1.06068917]),\n",
       " 'split1_test_score': array([-1.54737149]),\n",
       " 'split1_train_score': array([-1.07083168]),\n",
       " 'split2_test_score': array([-1.5504171]),\n",
       " 'split2_train_score': array([-1.08148427]),\n",
       " 'split3_test_score': array([-1.30197968]),\n",
       " 'split3_train_score': array([-1.13242]),\n",
       " 'split4_test_score': array([-1.51849097]),\n",
       " 'split4_train_score': array([-1.08478296]),\n",
       " 'std_fit_time': array([ 9.32839636]),\n",
       " 'std_score_time': array([ 0.18225519]),\n",
       " 'std_test_score': array([ 0.09690634]),\n",
       " 'std_train_score': array([ 0.02468763])}"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gs.cv_results_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
