{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Using TFIDF values to predict the class of genes\n",
    "\n",
    "### Table of Contents\n",
    "1. [Importing and formatting the data](#1)\n",
    "2. [Creating predictions](#1.1) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "import pandas as pd\n",
    "plt.style.use('ggplot')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Importing and formatting the data <a id=1></a>\n",
    "\n",
    "We need to do some data preprocessing at first.  Let us load the data where we will use 5x cross validation to evaluate the models.   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Michael\\Anaconda3\\envs\\tensorflow\\lib\\site-packages\\ipykernel\\__main__.py:1: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.\n",
      "  if __name__ == '__main__':\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(3321, 1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Text\n",
       "0  Cyclin-dependent kinases (CDKs) regulate a var...\n",
       "1   Abstract Background  Non-small cell lung canc...\n",
       "2   Abstract Background  Non-small cell lung canc...\n",
       "3  Recent evidence has demonstrated that acquired...\n",
       "4  Oncogenic mutations in the monomeric Casitas B..."
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_text = pd.read_csv('../data/training_text.csv',delimiter='\\|\\|',index_col=0,encoding='utf-8')\n",
    "df_text.columns=['Text']\n",
    "df_variants = pd.read_csv('../data/training_variants.csv',index_col=0)\n",
    "print(df_text.shape)\n",
    "df_text.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CBL</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBL</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene             Variation  Class\n",
       "ID                                     \n",
       "0   FAM58A  Truncating Mutations      1\n",
       "1      CBL                 W802*      2\n",
       "2      CBL                 Q249E      2\n",
       "3      CBL                 N454D      3\n",
       "4      CBL                 L399V      4"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_variants.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We now need to clean up the data, we see that there are several features we will want to remove from the dataset.  This task will be put in the process_text function  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df_text.Text[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"Cyclin-dependent kinases (CDKs) regulate a variety of fundamental cellular processes. CDK10 stands out as one of the last orphan CDKs for which no activating cyclin has been identified and no kinase activity revealed. Previous work has shown that CDK10 silencing increases ETS2 (v-ets erythroblastosis virus E26 oncogene homolog 2)-driven activation of the MAPK pathway, which confers tamoxifen resistance to breast cancer cells. The precise mechanisms by which CDK10 modulates ETS2 activity, and more generally the functions of CDK10, remain elusive. Here we demonstrate that CDK10 is a cyclin-dependent kinase by identifying cyclin M as an activating cyclin. Cyclin M, an orphan cyclin, is the product of FAM58A, whose mutations cause STAR syndrome, a human developmental anomaly whose features include toe syndactyly, telecanthus, and anogenital and renal malformations. We show that STAR syndrome-associated cyclin M mutants are unable to interact with CDK10. Cyclin M silencing phenocopies CDK10 silencing in increasing c-Raf and in conferring tamoxifen resistance to breast cancer cells. CDK10/cyclin M phosphorylates ETS2 in vitro, and in cells it positively controls ETS2 degradation by the proteasome. ETS2 protein levels are increased in cells derived from a STAR patient, and this increase is attributable to decreased cyclin M levels. Altogether, our results reveal an additional regulatory mechanism for ETS2, which plays key roles in cancer and development. They also shed light on the molecular mechanisms underlying STAR syndrome.Cyclin-dependent kinases (CDKs) play a pivotal role in the control of a number of fundamental cellular processes . The human genome containsgenes encoding proteins that can be considered as members of the CDK family owing to their sequence similarity with bona fide CDKs, those known to be activated by cyclins . Although discovered almosty ago , CDK10 remains one of the two CDKs without an identified cyclin partner. This knowledge gap has largely impeded the exploration of its biological functions. CDK10 can act as a positive cell cycle regulator in some cells  or as a tumor suppressor in others . CDK10 interacts with the ETS2 (v-ets erythroblastosis virus E26 oncogene homolog 2) transcription factor and inhibits its transcriptional activity through an unknown mechanism . CDK10 knockdown derepresses ETS2, which increases the expression of the c-Raf protein kinase, activates the MAPK pathway, and induces resistance of MCF7 cells to tamoxifen .Here, we deorphanize CDK10 by identifying cyclin M, the product of FAM58A, as a binding partner. Mutations in this gene that predict absence or truncation of cyclin M are associated with STAR syndrome, whose features include toe syndactyly, telecanthus, and anogenital and renal malformations in heterozygous females . However, both the functions of cyclin M and the pathogenesis of STAR syndrome remain unknown. We show that a recombinant CDK10/cyclin M heterodimer is an active protein kinase that phosphorylates ETS2 in vitro. Cyclin M silencing phenocopies CDK10 silencing in increasing c-Raf and phospho-ERK expression levels and in inducing tamoxifen resistance in estrogen receptor (ER)+ breast cancer cells. We show that CDK10/cyclin M positively controls ETS2 degradation by the proteasome, through the phosphorylation of two neighboring serines. Finally, we detect an increased ETS2 expression level in cells derived from a STAR patient, and we demonstrate that it is attributable to the decreased cyclin M expression level observed in these cells.Previous SectionNext SectionResultsA yeast two-hybrid (Y2H) screen unveiled an interaction signal between CDK10 and a mouse protein whose C-terminal half presents a strong sequence homology with the human FAM58A gene product [whose proposed name is cyclin M ]. We thus performed Y2H mating assays to determine whether human CDK10 interacts with human cyclin M (–C). The longest CDK10 isoform (P1) expressed as a bait protein produced a strong interaction phenotype with full-length cyclin M (expressed as a prey protein) but no detectable phenotype with cyclin D1, p21 (CIP1), and Cdi1 (KAP), which are known binding partners of other CDKs . CDK1 and CDK3 also produced Y2H signals with cyclin M, albeit notably weaker than that observed with CDK10 . An interaction phenotype was also observed between full-length cyclin M and CDK10 proteins expressed as bait and prey, respectively . We then tested different isoforms of CDK10 and cyclin M originating from alternative gene splicing, and two truncated cyclin M proteins corresponding to the hypothetical products of two mutated FAM58A genes found in STAR syndrome patients . None of these shorter isoforms produced interaction phenotypes ( and C and )..In a new window Download PPT.CDK10 and cyclin M form an interaction complex.  Schematic representation of the different protein isoforms analyzed by Y2H assays. Amino acid numbers are indicated. Black boxes indicate internal deletions. The red box indicates a differing amino acid sequence compared with CDK10 P1.  Y2H assay between a set of CDK proteins expressed as baits (in fusion to the LexA DNA binding domain) and CDK interacting proteins expressed as preys (in fusion to the B42 transcriptional activator). pEG202 and pJG4-5 are the empty bait and prey plasmids expressing LexA and B42, respectively. lacZ was used as a reporter gene, and blue yeast are indicative of a Y2H interaction phenotype.  Y2H assay between the different CDK10 and cyclin M isoforms. The amino-terminal region of ETS2, known to interact with CDK10 , was also assayed.  Western blot analysis of Myc-CDK10 (wt or kd) and CycM-V5-6His expression levels in transfected HEK293 cells.  Western blot analysis of Myc-CDK10 (wt or kd) immunoprecipitates obtained using the anti-Myc antibody. “Inputs” correspond toμg total lysates obtained from HEK293 cells coexpressing Myc-CDK10 (wt or kd) and CycM-V5-6His.  Western blot analysis of immunoprecipitates obtained using the anti-CDK10 antibody or a control goat antibody, from human breast cancer MCF7 cells. “Input” corresponds toμg MCF7 total cell lysates. The lower band of the doublet observed on the upper panel comigrates with the exogenously expressed untagged CDK10 and thus corresponds to endogenous CDK10. The upper band of the doublet corresponds to a nonspecific signal, as demonstrated by it insensitivity to either overexpression of CDK10 (as seen on the left lane) or silencing of CDK10 . Another experiment with a longer gel migration is shown in .Next we examined the ability of CDK10 and cyclin M to interact when expressed in human cells ( and E). We tested wild-type CDK10 (wt) and a kinase dead (kd) mutant bearing a D181A amino acid substitution that abolishes ATP binding . We expressed cyclin M-V5-6His and/or Myc-CDK10 (wt or kd) in a human embryonic kidney cell line (HEK293). The expression level of cyclin M-V5-6His was significantly increased upon coexpression with Myc-CDK10 (wt or kd) and, to a lesser extent, that of Myc-CDK10 (wt or kd) was increased upon coexpression with cyclin M-V5-6His . We then immunoprecipitated Myc-CDK10 proteins and detected the presence of cyclin M in the CDK10 (wt) and (kd) immunoprecipitates only when these proteins were coexpressed pair-wise . We confirmed these observations by detecting the presence of Myc-CDK10 in cyclin M-V5-6His immunoprecipitates . These experiments confirmed the lack of robust interaction between the CDK10.P2 isoform and cyclin M . To detect the interaction between endogenous proteins, we performed immunoprecipitations on nontransfected MCF7 cells derived from a human breast cancer. CDK10 and cyclin M antibodies detected their cognate endogenous proteins by Western blotting. We readily detected cyclin M in immunoprecipitates obtained with the CDK10 antibody but not with a control antibody . These results confirm the physical interaction between CDK10 and cyclin M in human cells.To unveil a hypothesized CDK10/cyclin M protein kinase activity, we produced GST-CDK10 and StrepII-cyclin M fusion proteins in insect cells, either individually or in combination. We observed that GST-CDK10 and StrepII-cyclin M copurified, thus confirming their interaction in yet another cellular model . We then performed in vitro kinase assays with purified proteins, using histone H1 as a generic substrate. Histone H1 phosphorylation was detected only from lysates of cells coexpressing GST-CDK10 and StrepII-cyclin M. No phosphorylation was detected when GST-CDK10 or StrepII-cyclin M were expressed alone, or when StrepII-cyclin M was coexpressed with GST-CDK10(kd) . Next we investigated whether ETS2, which is known to interact with CDK10  , is a phosphorylation substrate of CDK10/cyclin M. We detected strong phosphorylation of ETS2 by the GST-CDK10/StrepII-cyclin M purified heterodimer, whereas no phosphorylation was detected using GST-CDK10 alone or GST-CDK10(kd)/StrepII-cyclin M heterodimer ..In a new window Download PPT.CDK10 is a cyclin M-dependent protein kinase.  In vitro protein kinase assay on histone H1. Lysates from insect cells expressing different proteins were purified on a glutathione Sepharose matrix to capture GST-CDK10(wt or kd) fusion proteins alone, or in complex with STR-CycM fusion protein. Purified protein expression levels were analyzed by Western blots (Top and Upper Middle). The kinase activity was determined by autoradiography of histone H1, whose added amounts were visualized by Coomassie staining (Lower Middle and Bottom).  Same as in A, using purified recombinant 6His-ETS2 as a substrate.CDK10 silencing has been shown to increase ETS2-driven c-RAF transcription and to activate the MAPK pathway . We investigated whether cyclin M is also involved in this regulatory pathway. To aim at a highly specific silencing, we used siRNA pools (mix of four different siRNAs) at low final concentration (10 nM). Both CDK10 and cyclin M siRNA pools silenced the expression of their cognate targets ( and C and ) and, interestingly, the cyclin M siRNA pool also caused a marked decrease in CDK10 protein level ( and ). These results, and those shown in , suggest that cyclin M binding stabilizes CDK10. Cyclin M silencing induced an increase in c-Raf protein and mRNA levels ( and C) and in phosphorylated ERK1 and ERK2 protein levels , similarly to CDK10 silencing. As expected from these effects , CDK10 and cyclin M silencing both decreased the sensitivity of ER+ MCF7 cells to tamoxifen, to a similar extent. The combined silencing of both genes did not result in a higher resistance to the drug . Altogether, these observations demonstrate a functional interaction between cyclin M and CDK10, which negatively controls ETS2..In a new window Download PPT.Cyclin M silencing up-regulates c-Raf expression.  Western blot analysis of endogenous CDK10 and cyclin M expression levels in MCF7 cells, in response to siRNA-mediated gene silencing.  Western blot analysis of endogenous c-Raf expression levels in MCF7 cells, in response to CDK10 or cyclin M silencing. A quantification is shown in .  Quantitative RT-PCR analysis of CDK10, cyclin M, and c-Raf mRNA levels, in response to CDK10 (Upper) or cyclin M (Lower) silencing. **P ≤ 0.01; ***P ≤ 0.001.We then wished to explore the mechanism by which CDK10/cyclin M controls ETS2. ETS2 is a short-lived protein degraded by the proteasome . A straightforward hypothesis is that CDK10/cyclin M positively controls ETS2 degradation. We thus examined the impact of CDK10 or cyclin M silencing on ETS2 expression levels. The silencing of CDK10 and that of cyclin M caused an increase in the expression levels of an exogenously expressed Flag-ETS2 protein , as well as of the endogenous ETS2 protein . This increase is not attributable to increased ETS2 mRNA levels, which marginally fluctuated in response to CDK10 or cyclin M silencing . We then examined the expression levels of the Flag-tagged ETS2 protein when expressed alone or in combination with Myc-CDK10 or -CDK10(kd), with or without cyclin M-V5-6His. Flag-ETS2 was readily detected when expressed alone or, to a lesser extent, when coexpressed with CDK10(kd). However, its expression level was dramatically decreased when coexpressed with CDK10 alone, or with CDK10 and cyclin M . These observations suggest that endogenous cyclin M levels are in excess compared with those of CDK10 in MCF7 cells, and they show that the major decrease in ETS2 levels observed upon CDK10 coexpression involves CDK10 kinase activity. Treatment of cells coexpressing Flag-ETS2, CDK10, and cyclin M with the proteasome inhibitor MG132 largely rescued Flag-ETS2 expression levels ..In a new window Download PPT.CDK10/cyclin M controls ETS2 stability in human cancer derived cells.  Western blot analysis of endogenous ETS2 expression levels in MCF7 cells, in response to siRNA-mediated CDK10 and/or cyclin M silencing. A quantification is shown in .  Western blot analysis of exogenously expressed Flag-ETS2 protein levels in MCF7 cells cotransfected with empty vectors or coexpressing Myc-CDK10 (wt or kd), or Myc-CDK10/CycM-V5-6His. The latter cells were treated forh with the MG132 proteasome inhibitor. Proper expression of CDK10 and cyclin M tagged proteins was verified by Western blot analysis. (C and D) Western blot analysis of expression levels of exogenously expressed Flag-ETS2 wild-type or mutant proteins in MCF7 cells, in the absence of  or in response to  Myc-CDK10/CycM-V5-6His expression. Quantifications are shown in  C and D.A mass spectrometry analysis of recombinant ETS2 phosphorylated by CDK10/cyclin M in vitro revealed the existence of multiple phosphorylated residues, among which are two neighboring phospho-serines (at positionsand 225) that may form a phosphodegron  (–S8). To confirm this finding, we compared the phosphorylation level of recombinant ETS2wt with that of ETS2SASA protein, a mutant bearing alanine substitutions of these two serines. As expected from the existence of multiple phosphorylation sites, we detected a small but reproducible, significant decrease of phosphorylation level of ETS2SASA compared with ETS2wt , thus confirming that Ser220/Ser225 are phosphorylated by CDK10/cyclin M. To establish a direct link between ETS2 phosphorylation by CDK10/cyclin M and degradation, we examined the expression levels of Flag-ETS2SASA. In the absence of CDK10/cyclin M coexpression, it did not differ significantly from that of Flag-ETS2. This is contrary to that of Flag-ETS2DBM, bearing a deletion of the N-terminal destruction (D-) box that was previously shown to be involved in APC-Cdh1–mediated degradation of ETS2  . However, contrary to Flag-ETS2 wild type, the expression level of Flag-ETS2SASA remained insensitive to CDK10/cyclin M coexpression . Altogether, these results suggest that CDK10/cyclin M directly controls ETS2 degradation through the phosphorylation of these two serines.Finally, we studied a lymphoblastoid cell line derived from a patient with STAR syndrome, bearing FAM58A mutation c.555+1G>A, predicted to result in aberrant splicing . In accordance with incomplete skewing of X chromosome inactivation previously found in this patient, we detected a decreased expression level of cyclin M protein in the STAR cell line, compared with a control lymphoblastoid cell line. In line with our preceding observations, we detected an increased expression level of ETS2 protein in the STAR cell line compared with the control ( and ). We then examined by quantitative RT-PCR the mRNA expression levels of the corresponding genes. The STAR cell line showed a decreased expression level of cyclin M mRNA but an expression level of ETS2 mRNA similar to that of the control cell line . To demonstrate that the increase in ETS2 protein expression is indeed a result of the decreased cyclin M expression observed in the STAR patient-derived cell line, we expressed cyclin M-V5-6His in this cell line. This expression caused a decrease in ETS2 protein levels ..In a new window Download PPT.Decreased cyclin M expression in STAR patient-derived cells results in increased ETS2 protein level.  Western blot analysis of cyclin M and ETS2 protein levels in a STAR patient-derived lymphoblastoid cell line and in a control lymphoblastoid cell line, derived from a healthy individual. A quantification is shown in .  Quantitative RT-PCR analysis of cyclin M and ETS2 mRNA levels in the same cells. ***P ≤ Western blot analysis of ETS2 protein levels in the STAR patient-derived lymphoblastoid cell line transfected with an empty vector or a vector directing the expression of cyclin M-V5-6His. Another Western blot revealing endogenously and exogenously expressed cyclin M levels is shown in . A quantification of ETS2 protein levels is shown in .Previous SectionNext SectionDiscussionIn this work, we unveil the interaction between CDK10, the last orphan CDK discovered in the pregenomic era , and cyclin M, the only cyclin associated with a human genetic disease so far, and whose functions remain unknown . The closest paralogs of CDK10 within the CDK family are the CDK11 proteins, which interact with L-type cyclins . Interestingly, the closest paralog of these cyclins within the cyclin family is cyclin M . The fact that none of the shorter CDK10 isoforms interact robustly with cyclin M suggests that alternative splicing of the CDK10 gene  plays an important role in regulating CDK10 functions.The functional relevance of the interaction between CDK10 and cyclin M is supported by different observations. Both proteins seem to enhance each other’s stability, as judged from their increased expression levels when their partner is exogenously coexpressed  and from the much reduced endogenous CDK10 expression level observed in response to cyclin M silencing ( and ). CDK10 is subject to ubiquitin-mediated degradation . Our observations suggest that cyclin M protects CDK10 from such degradation and that it is the only cyclin partner of CDK10, at least in MCF7 cells. They also suggest that cyclin M stability is enhanced upon binding to CDK10, independently from its kinase activity, as seen for cyclin C and CDK8 . We uncover a cyclin M-dependent CDK10 protein kinase activity in vitro, thus demonstrating that this protein, which was named a CDK on the sole basis of its amino acid sequence, is indeed a genuine cyclin-dependent kinase. Our Y2H assays reveal that truncated cyclin M proteins corresponding to the hypothetical products of two STAR syndrome-associated FAM58A mutations do not produce an interaction phenotype with CDK10. Hence, regardless of whether these mutated mRNAs undergo nonsense-mediated decay (as suggested from the decreased cyclin M mRNA levels in STAR cells, shown in ) or give rise to truncated cyclin M proteins, females affected by the STAR syndrome must exhibit compromised CDK10/cyclin M kinase activity at least in some tissues and during specific developmental stages.We show that ETS2, a known interactor of CDK10, is a phosphorylation substrate of CDK10/cyclin M in vitro and that CDK10/cyclin M kinase activity positively controls ETS2 degradation by the proteasome. This control seems to be exerted through a very fine mechanism, as judged from the sensitivity of ETS2 levels to partially decreased CDK10 and cyclin M levels, achieved in MCF7 cells and observed in STAR cells, respectively. These findings offer a straightforward explanation for the already reported up-regulation of ETS2-driven transcription of c-RAF in response to CDK10 silencing . We bring evidence that CDK10/cyclin M directly controls ETS2 degradation through the phosphorylation of two neighboring serines, which may form a noncanonical β-TRCP phosphodegron (DSMCPAS) . Because none of these two serines precede a proline, they do not conform to usual CDK phosphorylation sites. However, multiple so-called transcriptional CDKs (CDK7, -8, -9, and -11) (to which CDK10 may belong; ) have been shown to phosphorylate a variety of motifs in a non–proline-directed fashion, especially in the context of molecular docking with the substrate . Here, it can be hypothesized that the high-affinity interaction between CDK10 and the Pointed domain of ETS2   would allow docking-mediated phosphorylation of atypical sites. The control of ETS2 degradation involves a number of players, including APC-Cdh1  and the cullin-RING ligase CRL4 . The formal identification of the ubiquitin ligase involved in the CDK10/cyclin M pathway and the elucidation of its concerted action with the other ubiquitin ligases to regulate ETS2 degradation will require further studies.Our results present a number of significant biological and medical implications. First, they shed light on the regulation of ETS2, which plays an important role in development  and is frequently deregulated in many cancers . Second, our results contribute to the understanding of the molecular mechanisms causing tamoxifen resistance associated with reduced CDK10 expression levels, and they suggest that, like CDK10 , cyclin M could also be a predictive clinical marker of hormone therapy response of ERα-positive breast cancer patients. Third, our findings offer an interesting hypothesis on the molecular mechanisms underlying STAR syndrome. Ets2 transgenic mice showing a less than twofold overexpression of Ets2 present severe cranial abnormalities , and those observed in STAR patients could thus be caused at least in part by increased ETS2 protein levels. Another expected consequence of enhanced ETS2 expression levels would be a decreased risk to develop certain types of cancers and an increased risk to develop others. Studies on various mouse models (including models of Down syndrome, in which three copies of ETS2 exist) have revealed that ETS2 dosage can repress or promote tumor growth and, hence, that ETS2 exerts noncell autonomous functions in cancer . Intringuingly, one of the very few STAR patients identified so far has been diagnosed with a nephroblastoma . Finally, our findings will facilitate the general exploration of the biological functions of CDK10 and, in particular, its role in the control of cell division. Previous studies have suggested either a positive role in cell cycle control  or a tumor-suppressive activity in some cancers . The severe growth retardation exhibited by STAR patients strongly suggests that CDK10/cyclin M plays an important role in the control of cell proliferation.Previous SectionNext SectionMaterials and MethodsCloning of CDK10 and cyclin M cDNAs, plasmid constructions, tamoxifen response analysis, quantitative RT-PCR, mass spectrometry experiments, and antibody production are detailed in SI Materials and Methods.Yeast Two-Hybrid Interaction Assays. We performed yeast interaction mating assays as previously described .Mammalian Cell Cultures and Transfections. We grew human HEK293 and MCF7 cells in DMEM supplemented with(vol/vol) FBS (Invitrogen), and we grew lymphoblastoid cells in RPMIGlutaMAX supplemented with(vol/vol) FBS. We transfected HEK293 and MCF7 cells using Lipofectamine(Invitrogen) for plasmids, Lipofectamine RNAiMAX (Invitrogen) for siRNAs, and Jetprime (Polyplus) for plasmids/siRNAs combinations according to the manufacturers’ instructions. We transfected lymphoblastoid cells by electroporation (Neon, Invitrogen). For ETS2 stability studies we treated MCF7 cellsh after transfection withμM MG132 (Fisher Scientific) forh.Coimmunoprecipitation and Western Blot Experiments. We collected cells by scraping in PBS (or centrifugation for lymphoblastoid cells) and lysed them by sonication in a lysis buffer containingmM β-glycerophosphate,mM p-nitrophenylphosphate,mM 3-(N-morpholino)propanesulfonic acid (Mops) (pH 7.2),mM EGTA,mM MgCl2,mM Na vanadate,mM NaF, 1mM phenylphosphate,Nonidet P-40, and a protease inhibitor mixture (Roche). We spun the lysatesmin at× g at°C, collected the supernatants, and determined the protein content using a Bradford assay. We performed the immunoprecipitation experiments onμg of total proteins, in lysis buffer. We precleared the lysates withμL of protein A or G-agarose beads, incubatedh°C on a rotating wheel. We addedμg of antibody to the supernatants, incubatedh°C on a rotating wheel, addedμL of protein A or G-agarose beads, and incubatedh°C on a rotating wheel. We collected the beads by centrifugations at× g at°C and washed three times in a bead buffer containingmM Tris (pH 7.4),mM NaF,mM NaCl,mM EDTA,mM EGTA,Nonidet P-40, and a protease inhibitor coktail (Roche). We directly added sample buffer to the washed pellets, heat-denatured the proteins, and ran the samples onBis-Tris SDS/PAGE. We transferred the proteins onto Hybond nitrocellulose membranes and processed the blots according to standard procedures. For Western blot experiments, we used the following primary antibodies: anti-Myc (Abcam ab9106, 1:2,000), anti-V5 (Invitrogen R960, 1:5,000), anti-tubulin (Santa Cruz Biotechnology B-7, 1:500), anti-CDK10 (Covalab pab0847p, 1:500 or Santa Cruz Biotechnology C-19, 1:500), anti-CycM (home-made, dilution 1:500 or Covalab pab0882-P, dilution 1:500), anti-Raf1 (Santa Cruz Biotechnology C-20, 1:1,000), anti-ETS2 (Santa Cruz Biotechnology C-20, 1:1,000), anti-Flag (Sigma F7425, 1:1,000), and anti-actin (Sigma A5060, 1:5,000). We used HRP-coupled anti-goat (Santa Cruz Biotechnology SC-2033, dilution 1:2,000), anti-mouse (Bio-Rad 170–6516, dilution 1:3,000) or anti-rabbit (Bio-Rad 172–1019, 1:5,000) as secondary antibodies. We revealed the blots by enhanced chemiluminescence (SuperSignal West Femto, Thermo Scientific).Production and Purification of Recombinant Proteins.GST-CDK10(kd)/StrepII-CycM. We generated recombinant bacmids in DH10Bac Escherichia coli and baculoviruses in Sf9 cells using the Bac-to-Bac system, as described by the provider (Invitrogen). We infected Sf9 cells with GST-CDK10- (or GST-CDK10kd)-producing viruses, or coinfected the cells with StrepII-CycM–producing viruses, and we collected the cellsh after infection. To purify GST-fusion proteins, we spunmL cells and resuspended the pellet inmL lysis buffer (PBS,mM NaCl,Nonidet P-40,mM NaF,mM β-glycerophosphate, andmM Na-vanadate) containing a protease inhibitor mixture (Roche). We lysed the cells by sonication, spun the lysatemin at× g, collected the soluble fraction, and added it to a 1-mL glutathione-Sepharose matrix. We incubatedh at°C, washed four times with lysis buffer, one time with kinase buffer A (see below), and finally resuspended the beads inμL kinase buffer A containing(vol/vol) glycerol for storage.6His-ETS2. We transformed Origami2 DE3 (Novagen) with the 6His-ETS2 expression vector. We induced expression withmM isopropyl-β-d-1-thiogalactopyranoside forh at°C. To purify 6His-ETS2, we spunmL cells and resuspended the pellet inmL lysis buffer (PBS,mM NaCl,mM Imidazole,mM DTT, andNonidet P-40) containing a protease inhibitor mixture without EDTA (Roche). We lysed the cells atbar using a cell disruptor and spun the lysatemin at× g. We collected the soluble fraction and added it toμL Cobalt beads (Thermo Scientific). Afterh incubation at°C on a rotating wheel, we washed four times with lysis buffer. To elute, we incubated beadsmin with elution buffer (PBS,mM imidazole, pH 7.6) containing the protease inhibitor mixture, spuns at× g, and collected the eluted protein.Protein Kinase Assays. We mixed glutathione-Sepharose beads (harboring GST-CDK10 wt or kd, either monomeric or complexed with StrepII-CycM),μM BSA,mM DTT,μM ATP,μCi ATP[γ-32P],μM histone H1, orμM 6His-ETS2 and added kinase buffer A (25 mM Tris·HCl,mM MgCl2,mM EGTA,mM DTT, andμM heparin, pH 7.5) up to a total volume ofμL. We incubated the reactionsmin at°C, added Laemli sample buffer, heat-denatured the samples, and ranBis-Tris SDS/PAGE. We cut gel slices to detect GST-CDK10 and StrepII-CycM by Western blotting. We stained the gel slices containing the substrate with Coomassie (R-250, Bio-Rad), dried them, and detected the incorporated radioactivity by autoradiography. We identified four unrelated girls with anogenital and renal malformations, dysmorphic facial features, normal intellect and syndactyly of toes. A similar combination of features had been reported previously in a mother–daughter pair1 (Tableand Supplementary Note online). These authors noted clinical overlap with Townes-Brocks syndrome but suggested that the phenotype represented a separate autosomal dominant entity (MIM601446). Here we define the cardinal features of this syndrome as a characteristic facial appearance with apparent telecanthus and broad tripartite nasal tip, variable syndactyly of toes 2–5, hypoplastic labia, anal atresia and urogenital malformations (a–h). We also observed a variety of other features (Table 1).  : Clinical and molecular characterization of STAR syndrome.  linical and molecular characterization of STAR syndrome. (a–f) Facial appearances of cases 1–3 (apparent telecanthus, dysplastic ears and thin upper lips; a,c,e), and toe syndactyly 2–5, 3–5 or 4–5 (b,d,f) in these cases illustrate recognizable features of STAR syndrome (specific parental consent has been obtained for publication of these photographs). Anal atresia and hypoplastic labia are not shown. (g,h) X-ray films of the feet of caseshowing only four rays on the left and delta-shaped 4th and 5th metatarsals on the right (h; compare to clinical picture in d).  Array-CGH data. Log2 ratio represents copy number loss of six probes spanning betweenandkb, with one probe positioned within FAM58A. The deletion does not remove parts of other functional genes.  Schematic structure of FAM58A and position of the mutations. FAM58A has five coding exons (boxes). The cyclin domain (green) is encoded by exons 2–4. The horizontal arrow indicates the deletion extending 5' in casewhich includes exonsandwhereas the horizontal line below exonindicates the deletion found in casewhich removes exonand some 3' sequence. The pink horizontal bars above the boxes indicate the amplicons used for qPCR and sequencing (one alternative exonamplicon is not indicated because of space constraints). The mutation 201dupT (case 4) results in an immediate stop codon, and the 555+1G>A and 555-1G>A splice mutations in cases5 andare predicted to be deleterious because they alter the conserved splice donor and acceptor site of intronrespectively.  Full size image (97 KB)  Table 1: Clinical features in STAR syndrome cases  Table- Clinical features in STAR syndrome cases  Full table  On the basis of the phenotypic overlap with Townes-Brocks, Okihiro and Feingold syndromes, we analyzed SALL1 , SALL4  and MYCN4 but found no mutations in any of these genes (Supplementary Methods online). Next, we carried out genome-wide high-resolution oligonucleotide array comparative genomic hybridization (CGH)5 analysis (Supplementary Methods) of genomic DNA from the most severely affected individual (casewith lower lid coloboma, epilepsy and syringomyelia) and identified a heterozygous deletion of 37.9–50.7 kb on Xq28, which removed exonsandof FAM58A (i,j). Using real-time PCR, we confirmed the deletion in the child and excluded it in her unaffected parents (Supplementary a online, Supplementary Methods and Supplementary Tableonline). Through CGH with a customized oligonucleotide array enriched in probes for Xq28, followed by breakpoint cloning, we defined the exact deletion size asbp (g.152,514,164_152,554,231del(chromosome X, NCBI Build 36.2); j and Supplementary Figs.online). The deletion removes the coding regions of exonsandas well as intron(2,774 bp),bp of intronandbp of 5' sequence, including the 5' UTR and the entire KRT18P48 pseudogene (NCBI gene ID 340598). Paternity was proven using routine methods. We did not find deletions overlapping FAM58A in the available copy number variation (CNV) databases.  Subsequently, we carried out qPCR analysis of the three other affected individuals (cases3 and 4) and the mother-daughter pair from the literature (casesand 6). In casewe detected a de novo heterozygous deletion of 1.1–10.3 kb overlapping exon(Supplementary b online). Using Xq28-targeted array CGH and breakpoint cloning, we identified a deletion ofbp (g.152,504,123_152,508,371del(chromosome X, NCBI Build 36.2); j and Supplementary ,3), which removedbp of intronall of exonincluding the 3' UTR, andbp of 3' sequence.  We found heterozygous FAM58A point mutations in the remaining cases (j, Supplementary upplementary Methods and Supplementary Table 1). In casewe identified the mutation 555+1G>A, affecting the splice donor site of intronIn casewe identified the frameshift mutation 201dupT, which immediately results in a premature stop codon N68XfsX1. In casesandwe detected the mutation 556-1G>A, which alters the splice acceptor site of intronWe validated the point mutations and deletions by independent rounds of PCR and sequencing or by qPCR. We confirmed paternity and de novo status of the point mutations and deletions in all sporadic cases. None of the mutations were seen in the DNA ofunaffected female controls, and no larger deletions involving FAM58A were found inunrelated array-CGH investigations.  By analyzing X-chromosome inactivation (Supplementary Methods and Supplementary Fig.online), we found complete skewing of X inactivation in casesand 3–6 and almost complete skewing in casesuggesting that cells carrying the mutation on the active X chromosome have a growth disadvantage during fetal development. Using RT-PCR on RNA from lymphoblastoid cells of case(Supplementary ), we did not find any aberrant splice products as additional evidence that the mutated allele is inactivated. Furthermore, FAM58A is subjected to X inactivation6. In casesandthe parental origin of the deletions could not be determined, as a result of lack of informative SNPs. Casethe mother of casegave birth to two boys, both clinically unaffected (samples not available). We cannot exclude that the condition is lethal in males. No fetal losses were reported from any of the families.  The function of FAM58A is unknown. The gene consists of five coding exons, and the 642-bp coding region encodes a protein ofamino acids. GenBank lists a mRNA length ofbp for the reference sequence (NM_152274.2). Expression of the gene (by EST data) was found inofadult tissues including kidney, colon, cervix and uterus, but not heart (NCBI expression viewer, UniGene Hs.496943). Expression was also noted inoflisted tumor tissues as well as in embryo and fetus. Genes homologous to FAM58A (NCBI HomoloGene: 13362) are found on the X chromosome in the chimpanzee and the dog. The zebrafish has a similar gene on chromosomeHowever, in the mouse and rat, there are no true homologs. These species have similar but intronless genes on chromosomes(mouse) and(rat), most likely arising from a retrotransposon insertion event. On the murine X chromosome, the flanking genes Atp2b3 and Dusp9 are conserved, but only remnants of the FAM58A sequence can be detected.  FAM58A contains a cyclin-box-fold domain, a protein-binding domain found in cyclins with a role in cell cycle and transcription control. No human phenotype resulting from a cyclin gene mutation has yet been reported. Homozygous knockout mice for Ccnd1 (encoding cyclin D1) are viable but small and have reduced lifespan. They also have dystrophic changes of the retina, likely as a result of decreased cell proliferation and degeneration of photoreceptor cells during embryogenesis7,Cyclin D1 colocalizes with SALL4 in the nucleus, and both proteins cooperatively mediate transcriptional repression9. As the phenotype of our cases overlaps considerably with that of Townes-Brocks syndrome caused by SALL1 mutations1, we carried out co-immunoprecipitation to find out if SALL1 or SALL4 would interact with FAM58A in a manner similar to that observed for SALL4 and cyclin D1. We found that FAM58A interacts with SALL1 but not with SALL4 (Supplementary Fig.online), supporting the hypothesis that FAM58A and SALL1 participate in the same developmental pathway.  How do FAM58A mutations lead to STAR syndrome? Growth retardation (all cases; Table 1) and retinal abnormalities (three cases) are reminiscent of the reduced body size and retinal anomalies in cyclin D1 knockout mice7,Therefore, a proliferation defect might be partly responsible for STAR syndrome. To address this question, we carried out a knockdown of FAM58A mRNA followed by a proliferation assay. Transfection of HEK293 cells with three different FAM58A-specific RNAi oligonucleotides resulted in a significant reduction of both FAM58A mRNA expression and proliferation of transfected cells (Supplementary Methods and Supplementary Fig.online), supporting the link between FAM58A and cell proliferation.  We found that loss-of-function mutations of FAM58A result in a rather homogeneous clinical phenotype. The additional anomalies in caseare likely to result from an effect of the 40-kb deletion on expression of a neighboring gene, possibly ATP2B3 or DUSP9. However, we cannot exclude that the homogeneous phenotype results from an ascertainment bias and that FAM58A mutations, including missense changes, could result in a broader spectrum of malformations. The genes causing the overlapping phenotypes of STAR syndrome and Townes-Brocks syndrome seem to act in the same pathway. Of note, MYCN, a gene mutated in Feingold syndrome, is a direct regulator of cyclin D2 ; thus, it is worth exploring whether the phenotypic similarities between Feingold and STAR syndrome might be explained by direct regulation of FAM58A by MYCN.  FAM58A is located approximatelyMb centromeric to MECP2 on Xq28. Duplications overlapping both MECP2 and FAM58A have been described and are not associated with a clinical phenotype in females12, but no deletions overlapping both MECP2 and FAM58A have been observed to date13. Although other genes between FAM58A and MECP2 have been implicated in brain development, FAM58A and MECP2 are the only genes in this region known to result in X-linked dominant phenotypes; thus, deletion of both genes on the same allele might be lethal in both males and females.\""
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def process_text(text):\n",
    "    text = re.sub(r'\\s+([0-9,.%]+)\\s+','',text) #remove numbers\n",
    "    text = re.sub(r'\\(ref[^)]+\\)','',text) #remove references ie \"(ref. 1)\"\n",
    "    text = re.sub('(:?[fF]ig)(:?s)?(:?\\.|(ure))(:?\\s*[a-zA-Z0-9]+)','',text) \n",
    "    text = re.sub(r'[([][0-9, ]*[)\\]]','',text) #remove references ie \"[1]\"\n",
    "    text = re.sub(r'\\([a-zA-Z]\\)','',text) #remove references ie \"[A]\"\n",
    "    return text\n",
    "process_text(df_text.Text[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\" Abstract Background  Non-small cell lung cancer (NSCLC) is a heterogeneous group of disorders with a number of genetic and proteomic alterations. c-CBL is an E3 ubiquitin ligase and adaptor molecule important in normal homeostasis and cancer. We determined the genetic variations of c-CBL, relationship to receptor tyrosine kinases (EGFR and MET), and functionality in NSCLC.  Methods and Findings  Using archival formalin-fixed paraffin embedded (FFPE) extracted genomic DNA, we show that c-CBL mutations occur in somatic fashion for lung cancers. c-CBL mutations were not mutually exclusive of MET or EGFR mutations; however they were independent of p53 and KRAS mutations. In normal/tumor pairwise analysis, there was significant loss of heterozygosity (LOH) for the c-CBL locus (22%, n\\u200a=\\u200a8/37) and none of these samples revealed any mutation in the remaining copy of c-CBL. The c-CBL LOH also positively correlated with EGFR and MET mutations observed in the same samples. Using select c-CBL somatic mutations such as S80N/H94Y, Q249E and W802* (obtained from Caucasian, Taiwanese and African-American samples, respectively) transfected in NSCLC cell lines, there was increased cell viability and cell motility.  Conclusions  Taking the overall mutation rate of c-CBL to be a combination as somatic missense mutation and LOH, it is clear that c-CBL is highly mutated in lung cancers and may play an essential role in lung tumorigenesis and metastasis.  Go to: Introduction In the US alone, each year approximately 219,400 people are diagnosed with lung cancers, out of which more than 145,000 of them succumb to the disease [1]. This number is roughly equivalent to the combined mortality rates of cancers of the breast, prostate, colon, liver, kidney and melanoma [1]. In addition the prognosis is usually poor and the five-year survival rate is less than 15%. There are also significant ethnic differences for lung cancer, and the outcome is worse for blacks compared to whites. Gender differences are also striking with women having significantly better prognosis as compared to men. There are a number of genetic alterations that can occur in lung cancer. As an example, in NSCLC, mutations in KRAS, p53, EGFR and MET have been identified. Many of these pathways, especially Receptor Tyrosine Kinases (RTKs) are controlled by c-CBL.  CBL (Casitas B-lineage lymphoma) is a mammalian gene located on human chromosome 11q23.3 [2] and is involved in cell signaling and protein ubiquitination [3]. CBL proteins belong to the RING finger class of ubiquitin ligases (E3) and there are three homologues c-CBL, CBL-b, CBL-3 [4]. The c-CBL and CBL-b genes are ubiquitously expressed with the highest levels in hematopoietic tissues [5]. c-CBL consists of four regions encoding for functionally distinct protein domains: the N-terminal tyrosine kinase binding (TKB) domain, the linker region, the catalytic RING finger domain, the proline-rich region and the c-terminal ubiquitin-associated (UBA) domain that also overlaps with a leucine-zipper (LZ) domain [3]. Both TKB and RING finger domains are essential for ligand-induced ubiquitination of RTKs [6], [7], [8], [9]. The RING finger domain is required for the recruitment of E2 ubiquitin-conjugating enzymes. The TKB domain includes four-helix bundle (4H), a calcium-biding EF hand, and a modified SH2 domain, which binds to phosphotyrosine residues [3], [10], [11], [12]. In addition, the proline-rich region of c-CBL can associate with the SH3 domain of Grb2, which can indirectly recruit c-CBL to RTKs via the GRB2 adaptor protein [7], [13], [14].  c-CBL also binds to EGFR and acts as the E3 that targets EGFR for ubiquitination and degradation. Furthermore, CBL desensitizes EGF signaling and opposes cellular proliferation induced by EGF [15]. EGF activation also appears to activate the tyrosine kinase SRC, which phosphorylates c-CBL and in turn activates the ubiquitination and degradation of EGFR [16], [17], [18]. A recent study shows that defective endocytosis of EGFR is characterized by a deletion mutant and the point mutation L858R, whereby its association with c-CBL and subsequent ubiquitination are impaired [19]. Recently, the first human c-CBL mutations were reported in acute myeloid leukemia (AML) patients [20]. The mutation R420Q inhibits FMS-like tyrosine kinase 3 (FLT3) internalization and ubiquitination [20].  Not only can E3 activity be important in oncogenesis, c-CBL has a dual but separate function as a signal transduction molecule. We have previously shown that c-CBL is important in binding CRKL and BCR/ABL in hematopoietic cells. Also, it can bind and modulate functions of cytoskeleton by binding to proteins like talin and paxillin. The TKB domain is important in binding to a number of molecules, and they then function in signal transduction.  Given the critical role of CBL in normal homeostasis and cancer, we hypothesized that it might be mutated in lung cancers. In this study, we report novel c-CBL somatic mutations S80N/H94Y, Q249E and W802* in Caucasian, Taiwanese and African-American lung cancer patients, respectively. Expressing these mutations in NSCLC cell lines lead to increased proliferation and cell motility. We show that c-CBL mutations occur with or without MET or EGFR mutations but are mutually exclusive of a LOH at the c-CBL locus. Additionally, c-CBL LOH is associated with either MET or EGFR mutations. We thus hypothesize that c-CBL mutations might contribute to the oncogenic potential of MET and EGFR in lung cancer.  Go to: Methods Ethics Statement  Written consent on all research on human subjects has been obtained from the Institutional Review Board, University of Chicago and covers all research performed in the laboratory. The following is their contact information: Institutional Review Board, The University of Chicago, McGiffert Hall, 5751 S. Woodlawn Ave., 2nd floor, Chicago, IL 60637. Written informed consents were received from all patients whose tissue samples were used for this study.  Tissue Samples  Lung cancer tissue and paired adjacent normal lung tissues were obtained from 50 Caucasian, 29 African-Americans and 40 Taiwanese NSCLC patients who were recruited at the University of Chicago Hospital (Chicago, USA) (Caucasian and African-American patients) and Taipei Veterans General Hospital of Taiwan (Taiwanese patients) after obtaining appropriate Institutional Review Board permission and informed consent from the patients. Out of 119 samples, 77 were men, 38 were women and 4 were unknown with age at diagnosis ranging from 47 to 90 years. In terms of tumor types, 53 were adenocarcinoma, 32 were squamous cell carcinoma and 34 were large cell carcinoma. 49 were stage I, 14 were stage II, 34 were stage III, and 13 were stage IV (Table S1).  Cell Culture  Human non-small cell lung carcinoma cells A549 and H358 were maintained in DMEM and RPMI-1640, respectively. Human embryonic kidney 293T cells were cultured in DMEM. Media were supplemented with 10% fetal bovine serum, 100 units/ml of penicillin, and 100 µg/ml of streptomycin (Invitrogen, Carlsbad, CA). Cells were cultured at 37°C in a humidified incubator containing 5% CO2.  c-CBL Gene Mutational Analysis  Exons 2 to 16 of c-CBL gene were individually amplified by polymerase chain reaction (PCR). Primers are listed in Table S2. PCR conditions were 1 cycle of 95°C for 5 minutes; 35 cycles of 94°C for 30 seconds, 58°C for 30 seconds and 72°C for 2 minutes; and one cycle of 72°C for 10 minutes. PCR products were treated with ExoSAP-IT (USB Corporation, Cleveland, OH) and sequenced by Big-Dye Terminator Chemistry (Applied Biosystems, Foster City, CA). Sequencing was performed on the forward coding strand with confirmation of c-CBL alterations performed by sequencing the reverse strand as well. Chromatograms were analyzed for mutations using Mutation Surveyor v2.61 (Softgenetics, State College, PA).  Plasmid Constructs and Site-Directed Mutagenesis  The wild-type c-CBL cDNA insert was subcloned into the pAlterMax expression vector using XhoI and SalI restriction enzyme sites (Promega, Madison, WI). Using this parental plasmid pAlterMax-c-CBL, the TKB domain double mutation (S80N/H94Y), the point mutation (Q249E), and the C-terminal point mutation W802* of c-CBL were created using the following primers: 5′-GCTGGCGCTAAAGAATAACCCACCTTATATCTTAGAC-3′ and 5′-CTACCAGATACCTACCAGTATCTCCGTACTATCTTGTC-3′ for the double mutation S80N/H94Y; 5′-CTTTACCCGACTCTTTGAGCCCTGGTCCTCTTTGC-3′ for Q249E, and 5′-CAGCTCCTCCTTTGGCTGATTGTCTCTGGATGGTGATC-3′ for W802* along with their complementary primers using the QuickChange Site-Directed Mutagenesis XL kit (Stratagene, La Jolla, CA) according to the manufacturer's instructions. The constructs were confirmed for the point mutations by standard DNA sequencing of both strands.  Loss of Heterozygosity (LOH) Analysis  Five microsatellites on chromosome 11 (3 on 11q at or within 200 kb up or downstream of the c-CBL gene and 2 control markers on 11p) were selected for analysis (Table S3). Established microsatellite markers and respective primer sequences were selected from the GeneLoc database (http://genecards.weizmann.ac.il/geneloc/index.shtml, Weizmann Institute of Science, Rehovot, Israel). Primers were custom designed and each forward primer was fluorescently labeled at the 5′ end with FAM, PET, NED, or VIC (Applied Biosystems). Primer annealing temperatures and duplex scores were evaluated with NIST Primer Tools (http://yellow.nist.gov:8444/dnaAnalysis/primerToolsPage.do; National Institute of Standards and Technology, Gaithersburg, MD). Primers were verified by performing PCR with control DNA (isolated from TK6 cells) and resolving the products on agarose gels. Bands were visualized with an UV transilluminator. Genomic DNA was extracted from tumor samples and paired normal lung tissue. Primers were grouped into multiplex combinations shown in Table S4. Marker D11S929 served as an internal control to check for consistency in PCRs and of peaks from capillary electrophoresis. Multiplex PCRs were carried out in a volume of 10 µL that contained 1 µL genomic DNA (20–50 ng), 0.5 µM of each primer (1.0 µM total for each primer pair), 400 µM dNTPs, 1X PCR buffer containing MgCl2, and 0.2 U Taq DNA polymerase. PCR was performed on the ABI GeneAmp 9700 PCR System under the following conditions: 5 min at 94°C; 30 cycles of 30 sec at 94°C, 1 min at 60°C, 1 min at 72°C; and 5 min at 72°C. The PCR products were separated by capillary electrophoresis on an ABI 3130XL DNA Analyzer. Chromatograms were analyzed with Peak Scanner 1.0 and GeneMapper 3.7 software (Applied Biosystems) for allelic alterations. The area of the peaks produced by the DNA PCR products was quantified for each allele. The ratio of the allelic areas was calculated for each tumor and paired normal DNA sample. When the qLOH (allelic ratio for the tumor peaks divided by the allelic ratio of paired normal sample) was ≤0.5 or ≥2.0 for c-CBL and at least one other 11q marker in at least two separate experiments, the sample was considered as having an allelic imbalance and interpreted as LOH. Samples were evaluated in at least two separate experiments and samples showing prospective LOH at c-CBL repeated a third time which included a new control marker at the BAX locus (data not shown) on chromosome 19 to verify integrity of sample DNA.  Transfection of c-Cbl Constructs  The A549 cell line was transfected using the Fugene HD (Roche, Nutley, NJ) reagent according to the manufacturer's instructions. Eight µg of plasmid DNA, containing either no insert (empty vector), wild-type c-CBL, S80N/H94Y c-CBL, Q249E c-CBL or W802* CBL was used for transfection in a 6-well culture plate. Cells were harvested 48 h after transfection and analyzed for expression.  c-CBL Knockdown  c-CBL knockdown was performed using lentiviral transduction using MISSION lentiviral transduction particles (Sigma-Aldrich, St. Louis, MO) as per manufacturer's instructions. Briefly, 1×105 H358 cells/well were seeded in 6-well plates and infected the following day with c-CBL lentiviral shRNA constructs. To generate stable c-CBL knockdown cell lines, cells were selected for 2 days with 1 µg/ml puromycin. c-CBL levels were determined using whole cell lysates by immunoblotting with anti-CBL antibody (Santa Cruz Biotechnologies, Santa Cruz, CA).  Cell Viability Assay  Cells were transfected as described above in the transfection assay. Forty-eight hours after transfection, viability of cells was assessed using Trypan Blue exclusion.  Wound Healing Assay  A549 cells were seeded in 6-well plates and cultured for 48 h until 100% confluent. The medium was then changed and the cells were transfected as described in the transfection assay. Twelve hours after transfection, a straight scratch was made across the cell layer using a 1 ml pipette tip. The cells were then gently washed with 1× PBS to remove cellular debris and the media was replaced. Photographs were taken of the wound region every 12 h until 48 h.  Western Blot Analysis  Forty eight hours after transfection, cells were collected and washed twice in 1X PBS, then lysed in ice-cold lysis buffer (0.5M Tris-HCl with pH 7.4, 1.5 M NaCl, 2.5% deoxycholic acid, 10 mM EDTA, 10% NP-40, 0.5 mM DTT, 1 mM phenylmethylsulfonyl fluoride, 5 µg/mL leupeptin, and 10 µg/mL aprotinin) for 5 minutes. The lysate was centrifuged at 13,000 rpm for 20 minutes at 4°C, and protein content of the supernatant was measured. Total cell lysates (50 µg/well) were separated by SDS-PAGE electrophoresis and the gels transferred onto nitrocellulose membranes (Whatman, Piscataway, NJ). Membranes were blocked with 5% non-fat dry milk in phosphate-buffered saline containing Tween-20 (PBST) (1X PBS, 0.1% Tween-20) for 1 h at room temperature and incubated with the appropriate primary antibody at 4°C overnight. Membranes then were washed three times with PBST and probed with appropriate horseradish peroxidase (HRP)-conjugated secondary antibody for 1 h at room temperature. The membranes were again washed three times in PBST and bands were visualized using Western blot chemiluminescence reagent (BioRad, Valencia, CA) on a Chemidoc Gel documentation system (BioRad, Valencia, CA). Antibodies were obtained from Santa Cruz Biotechnologies and used at the following dilutions (c-CBL, 1∶5000; c-MET, 1∶5000; EGFR, 1∶5000; ubiquitin, 1∶1000; HA, 1∶5000 and β-actin, 1∶10,000).  Flow Cytometry  Cell cycle analysis was carried out by flow cytometry. Approximately 2×106 cells were grown in media containing 10% FBS. Cells were harvested by trypsin/EDTA treatment, washed with 1X PBS three times and fixed with ice-cold 70% ethanol for 2 h. Cells were washed again with cold PBS and stained with a solution containing 25 µg/mL propidium iodide, 200 µg/mL RNase A, and 0.1% Triton X-100 for 30 minutes in the dark. Cell cycle analysis was performed using a Guava PCA-96 flow cytometer (Guava Technologies, Millipore, Billerica, MA).  Ubiquitin Ligase Activity  293T cells were maintained in culture in DMEM supplemented with 10% FBS and 1% penicillin (100 units/mL) and streptomycin (100 µg/mL) were transfected with 0.2 µg EGFR-pcDNA3 and 2 µg HA-tagged c-CBL constructs as indicated using calcium phosphate according to manufacturer's protocol (Profection, Promega, Madison, WI). Twenty-four hours post-transfection, cells were starved overnight in DMEM supplemented with 0.5% FBS, and then treated with or without EGF (100 ng/ml) for 15 min. The cells were collected and washed two times in ice-cold PBS containing 0.2 mM sodium orthovanadate then lysed in ice-cold lysis buffer (10 mM Tris HCl, pH 7.5, 150 mM NaCl, 5 mM EDTA, 1% Triton X100, 10% Glycerol, 2 mM sodium orthovanadate and protease inhibitors). Lysates were cleared of debris by centrifugation at 16,000 g for 10 min at 4°C. EGFR immunoprecipitations were performed on 200 µg of cleared lysate using 250 ng of rabbit-anti-EGFR and Protein A/G Plus Sepharose overnight at 4°C. Precipitations were washed 5 times in lysis buffer before boiling in Laemmli buffer. Elutions were immunoblotted with anti-ubiquitin and EGFR. Twenty micrograms of cleared lysate were immunoblotted for each of the c-CBL constructs using anti-HA.  Statistical Analysis  Mutation rates between different groups were compared using Fisher's exact test. For continuous variables, group comparisons were performed using analysis of variance (ANOVA) followed by Sidak's adjustment for multiple comparisons. Experiments involving repeated measurements over time were analyzed using repeated measures ANOVA with the Greenhouse-Geisser adjustment to the degrees of freedom. Analyses were conducted using STATA (v10.1) software (Stata Corporation, College Station, TX).  Go to: Results c-CBL Gene Mutations in Lung Cancer  To investigate the role of c-CBL in lung cancer, we analyzed its genomic DNA in tumor and paired normal samples drawn from multiple ethnicities. The lung tumor samples represented Caucasians (n\\u200a=\\u200a50), African-Americans (n\\u200a=\\u200a29), and Taiwanese (n\\u200a=\\u200a40) lung cancer patients. We designed 12 pairs of primers to sequence the coding region of c-CBL gene that spans exons 2 to 16 (Table S2). We identified 8 unique somatic mutations in c-CBL exons among 8 different patients. A variation L620F, a known SNP (rs2227988) in exon 11 was also detected. Importantly, the eight novel non-synonymous mutations were confirmed by sequencing both strands of c-CBL genomic DNA obtained from lung tumor samples (Table 1). Moreover, none of the 8 mutations were detected in the corresponding normal tissue, indicating that these were somatic mutations. Four synonymous single nucleotide variations (SNVs) were also identified but were not used further in this study.  Table 1 Table 1 c-CBL mutation analysis in 119 lung cancer patient tumor tissues. Three of the 8 novel non-synonymous mutations were located in the TKB (tyrosine kinase binding) domain (S80N, H94Y, and Q249E), one in the RING finger domain (V391I), one in the proline-rich region (72515_72517 del ATG), and three in the C-terminal region (W802*, R830K, and A848T) of the c-CBL protein (Figure 1A and Figure S1). In Figure 1B, we show model chromatograms of representative samples.  Figure 1 Figure 1 c-CBL mutations and LOH in non-small cell lung cancer. 11q LOH of c-CBL Gene  Paired lung tumor and normal lung tissue samples from Taiwanese patients (n\\u200a=\\u200a37) were investigated for LOH. Eight (21.6%) showed LOH at the c-CBL locus on chromosome 11 while 29 samples (78.4%) revealed normal allelic contribution at the microsatellite markers (Figures 1C and D).  c-CBL Mutations in Different Ethnic Groups  The c-CBL double mutant S80N/H94Y was found in the same patient and the overall mutation rate for c-CBL in lung tumors was 6.7% (8/119). The frequency of c-CBL mutation was highest in large cell carcinoma (14.7%; 5 of 34 patients) followed by squamous carcinoma (6.3%; 2 of 32 patients) and the least was observed in adenocarcinoma (AD) (1.8%; 1 of 53 patients), although these rates were not statistically significant (p\\u200a=\\u200a0.292). Mutation rates were 6.0% among Caucasians (0 of 20 in AD; 0 of 10 in SQ; and 3 of 20 in LC), 13.8% in African-Americans (1 of 10 in AD; 1 of 10 in SQ; and 2 of 9 in LC), and 2.5% (0 of 23 in AD; 1 of 12 in SQ; and 0 of 5 in LC) in the Taiwanese population. Additionally two Taiwanese patients with lung cancer (one squamous and one adenocarcinoma) had the known SNP L620F. Ethnic differences were not statistically significant, however the power to detect differences was low.  Mutations in MET and EGFR Can Be Co-Associated with c-CBL Alterations  Since East Asians with lung cancer have a higher frequency of EGFR and MET mutations in lung tumors [21], [22], we also determined mutations in EGFR and MET in the same Taiwanese cohort samples and compared the results with the observed c-CBL alterations (LOH and/or mutations). In the 37 samples tested, we did not find any overlap between c-CBL mutations and c-CBL LOH (Figure 2). Of the three c-CBL mutants (including the known L620F SNP, rs2227988), one of the samples had a MET mutation (N375S) and the other had an EGFR mutation (L858R). Among the 8 samples that had a LOH at the c-CBL locus 5 had an additional mutation in MET (N375S) and 2 had an EGFR exon 19 deletion. Twenty-six samples had neither c-CBL mutation nor c-CBL LOH (3 patients had a c-CBL mutation but no c-CBL LOH). Among these 26 samples 9 had a MET mutation (8 N375S, 1 L211W), 13 had an EGFR mutation (7 exon 9 deletion, 6 L858R) and 4 had no other MET or EGFR mutation. Thus the rate of MET or EGFR mutations among patients with LOH at the c-CBL locus (7 of 8) was similar to that seen in patients without c-CBL mutation or LOH (22 of 26 patients) (p\\u200a=\\u200a0.99). These 4 patients with no identifiable mutation in c-CBL, MET or EGFR represented 10.8% of the 37 patients analyzed in the Taiwanese patient cohort. Conversely, 89.2% Taiwanese lung cancer patients have an identifiable mutation in either c-CBL, MET or EGFR or a combination of the three genes (Figure 2). Additionally, we determined p53 and KRAS mutations in these Taiwanese cohorts. Two p53 and 1 KRAS mutation were detected. The single KRAS mutation overlapped with one p53 mutation. This patient also had the EGFR exon 19 deletion but had no c-CBL mutation. The other p53 mutation sample had a c-CBL LOH with concurrent MET N375S mutation. Thus, in the Taiwanese samples analyzed, p53/KRAS mutations and c-CBL mutations were mutually exclusive (data not shown).  Figure 2 Figure 2 c-CBL mutations and relationship to MET and EGFR mutations in lung cancer. Cellular Functions of c-CBL Alterations in the Context of Lung Tumorigenesis  A. E3 activity is intact in the mutant c-CBL proteins To investigate whether the different c-CBL mutations affect the E3 activity, EGFR was chosen as a model substrate for c-CBL E3 function. All of the c-CBL mutants tested enhanced ubiquitination of the activated EGFR similar to the wild-type c-CBL protein. This result demonstrates that the catalytic activity of the c-CBL mutants is not impaired when EGFR was the substrate. (Figure 3A).  Figure 3 Figure 3 Ubiquitination, viability, expression and cell cycle analysis of various c-CBL mutants. B. Effect on lung cancer cell viability The effect of a representative c-CBL mutant from each of the three ethnic backgrounds on lung cancer cell viability in cell lines was determined. S80N/H94Y double mutation, Q249E, and W802* were identified in lung tumor samples obtained from a Caucasian, a Taiwanese and an African-American, respectively. As described in methods, the c-CBL wild-type (WT) and the above three mutants were expressed after cloning them into pAlterMax vector in A549 cells. These cells express relatively low basal levels of endogenous c-CBL (data not shown). Transfection efficiency was comparable between different groups and the number of cells transfected with c-CBL wild-type construct was about 70% compared to control cells that were transfected with the empty vector. Cells transfected with S80N/H94Y, Q249E and W802* c-CBL mutant constructs resulted in increased number of viable cells that was 132.3%, 120.8% and 147.9% higher respectively, relative to the empty vector control transfected cells and significantly different from the wild-type construct (p\\u200a=\\u200a0.022, p\\u200a=\\u200a0.049, and p\\u200a=\\u200a0.008, respectively) (Figure 3B). Relative levels of c-CBL protein in whole cell lysates prepared from samples obtained from a parallel experiment was determined. The c-CBL protein levels in samples representing untransfected and empty vector transfected cells were comparable and those representing the c-CBL WT and the three c-CBL mutants were comparable (Figure 3C).  C. Effect on cell cycle To investigate if the increases in cell viability in different c-CBL mutants are due to increased cellular proliferation, a cell cycle analysis was performed. A549 cells were transfected with the c-CBL WT or the three different mutants: S80N/H94Y, Q249E and W802*. The empty vector transfectant was used as a control. Forty-eight hours after transfection cell cycle analysis was performed as described in materials and methods. There was no significant change in the subG1, G1 or the S phase of the cell cycle among the different mutants compared to the WT construct (p\\u200a=\\u200a0.64, p\\u200a=\\u200a0.40, and p\\u200a=\\u200a0.28, respectively). The G2/M phase of the cell cycle showed an increase in cell numbers for the three mutants, S80N/H94Y, Q249E and W802*, when compared to the WT but again the difference was not statistically significant (p\\u200a=\\u200a0.25) (Figure 3D).  D. Effect on cell motility To investigate the effect of the expression of the above three c-CBL mutants on cell migration, we carried out wound healing assay as described in materials and methods. The closing of the scratch or the wound was monitored at 0, 12, 24, 36, and 48 h. (Figure 4A). In all the samples, that represented cells transfected with mutants, the wound gap was much smaller than that seen in the sample that represented cells transfected with c-CBL WT (p<0.001). We also determined the rate of wound closure for all the five groups. At 48 h, wild-type c-CBL transfectants showed 61.1% open wound while the S80N/H94Y, Q249E and W802* mutants showed 18.7%, 23.9% and 34.3% open wound respectively (p<0.001) (Figure 4B).  Figure 4 Figure 4 c-CBL mutations affect wound healing in A549 cells. E. c-CBL knockdown increases cell viability It is hypothesized that the LOH seen in our samples could lead to decreased expression of c-CBL. Thus we tested the effect of c-CBL knockdown in lung cancer cells. Compared to A549, H358 lung cancer cells express relatively high levels of endogenous c-CBL (data not shown). c-CBL expression was knocked down using lentiviral construct that expressed c-CBL specific shRNA and compared the results with those that were transduced with scrambled shRNA and the results are shown in Figure 5. We identified several clones that revealed varying degrees of c-CBL knockdown showing different sets of c-CBL lentiviral shRNA knockdown efficiency (Figure 5A). Of all the clones tested, Clone 27 was chosen for further experiments. Equal amount of cells were seeded in a 6-well plate and the cell proliferation was measured at various times and the results are depicted in Figure 5B. As expected, number of cells increased in a time dependent fashion from 100 to 190% relative to scrambled shRNA as control in a span of 48 h (p\\u200a=\\u200a0.0002) (Figure 5B). The cell cycle phases in H358 cells that were knocked down with c-CBL shRNA were looked at and compared with the scrambled shRNA. There were no discernable differences between these two constructs in the different phases of the cell cycle (data not shown).  Figure 5 Figure 5 Knockdown of c-CBL using an shRNA increases cell proliferation. Go to: Discussion Our results demonstrate that c-CBL is somatically mutated (or has LOH) in lung cancers, and can significantly contribute to enhanced cell viability and motility. There was also a high prevalence of LOH with respect to c-CBL in lung tumors that harbored MET or EGFR mutation.  In the present study, we have demonstrated the occurrence of c-CBL mutations in lung cancer patients, especially with different ancestral variations. Mutations in c-CBL have been recently reported in juvenile myelomonocytic leukemia and myeloid malignancies. In the AML study, the mutation R420Q located in the junction of the RING finger and the linker region inhibited FMS-like tyrosine kinase 3 (FLT3) internalization and ubiquitination [20], thus contributing to the gain-in-function for the RTK. In addition, mutations such as H398Y, C384R, and L380P mapped to the RING finger domain and the linker region of c-CBL that is required for its E3 activity [23], [24], [25], [26], [27]. Additionally, homozygous mutations in the RING finger domain of the c-CBL gene were described as a result of acquired Uniparental Disomy (UPD) [26]. It is important to note that our results indicate LOH at 11q23 locus and these are mutually exclusive from missense mutations of c-CBL. The somatic mutations were all heterozygous. The mutations in AML led to abrogation of the E3 activity leading to prolonged RTK activation. In addition mutants located on the linker region surrounding the RING finger domain exhibited enhanced AKT signaling in response to cytokine stimulation [26]. In addition, it was shown in NH3T3 cells, that neither mutations in the RING finger nor the linker region causes transformation, however while certain mutations perturbs the ubiquitination, others affect receptor recycling and prolong kinase activity [28].  We report here c-CBL mutations that mapped not only to the RING finger domain, but also to the TKB domain, proline-rich domain and the C-terminal region, but none mapped to the linker region as reported in the AML studies described above [23], [24], [25], [26], [29]. In addition, 8 mutants that we detected were found in different ethnic backgrounds. For example, S80N/H94Y, Q249E, W802* were detected in Caucasians, Taiwanese and African-Americans, respectively. The results point out not only the difference between lung cancer and other cancers, but also genetic polymorphism among different races in the same cancer. Interestingly, there is a large disparity between African-American and other ethnic populations with lung cancer [30]. We have previously shown that there was a low frequency of EGFR and MET mutation in African-Americans as compared with Taiwanese and Caucasians [31]. In this study, the number of African-American samples analyzed was relatively fewer and we found 3 mutations that are unique to this ethnicity. It would now behoove us to further study the genetic alterations that can occur and determine the targeted therapeutics for African-Americans. Our results provide evidence of the importance of c-CBL in tumorigenesis and potential signaling. Our prediction, based on the AML data, would be that the V391I RING finger domain mutation would affect the E3 activity. Also, it will be important to determine the binding partners for c-CBL in the TKB domain and proline rich domain mutations. It has previously been shown that the TKB domain can bind to growth factor receptors and it will be important to determine the cross-binding of these mutants to MET and or EGFR. It would also be important in the future to look at fluorescence in situ hybridization/copy number changes in c-CBL in lung cancer.  c-CBL plays an important role in down regulating RTK-mediated signaling through K63 poly-ubiquitination and subsequent downregulation of RTKs followed by lysosomal degredation [3]. Mono-ubiquitination or ubiquitinated with K63-linked chains of substrates by c-CBL may lead to enhancement of biological and biochemical functions (reviewed in Hermann et al, 2007 [32]). The mutations that we analyzed in our studies all point out to the fact that the E3 activity of c-CBL on EGFR is intact; the EGFR levels in the various mutants remain same (Figure S2). Multiple kinases, both RTKs and non-RTKs could be acted upon by c-CBL, including ERBs, PDGFR, FMS, MET, c-Kit, VEGFR, FLT-1, RON, FGFR, IR, as well as SYK, FYN, LCK, FGR, LYN and c-ABL [3]. In lung cancers the relevant substrates of c-CBL in terms of degradation or signal transduction are yet to be identified.  The observation that c-CBL somatic mutations, especially S80N/H94Y, Q249E and W802* showed increased cell viability and cell motility that is in agreement with the physiological role for D-cbl in the regulation of apoptosis and differentiation identified in Drosophila is very significant [33]. It has been previously shown that activating c-CBL mutation downregulates EGFR signaling and decreases cellular proliferation and migration in breast cancer cell lines [34]. Although the role of c-CBL in the negative regulation of RTKs is well substantiated, thereby suggesting that it is a natural tumor suppressor, studies in cancer cells have revealed both tumor suppressor and tumor promoting activities depending on the type of c-CBL mutation and the number of alleles at the c-CBL locus [24]. In agreement with the above, the three c-CBL mutants described here appear to have tumor growth and metastasis promoting properties. Although these mutants are outside of the RING finger or the linker region of c-CBL, their downstream effects are significant so as to cause increased proliferation and migration, but the substrate affected by these mutations are not known yet. This raises the possibility that some of the cellular functions of c-CBL are independent of its ubiquitin-ligase activity, an area that we are currently investigating. The oncogenic nature of RTKs, addiction of cancers to growth signals and given the clustering of c-CBL, EGFR and MET mutations, it is possible that the transforming effect of c-CBL mutations is most likely a combinatorial effect of the three. We also show that LOH for c-CBL was found in a significant number of samples that harbored MET or EGFR mutations. The fact that about 7% of lung tumor samples are likely to have c-CBL mutations and an additional 22% are likely to harbor c-CBL-related LOH, makes c-CBL a highly mutated molecule in lung cancer. Since LOH alone is not enough to cause a transforming event [35], [36], [37], associated mutation in the MET or EGFR locus or yet another RTK discussed above may play a role in carcinogenesis. We predict that this LOH in c-CBL results in haploinsufficiency that downplays RTK ubiquitination leading to hyperactivity of the RTKs. However, whether this is sufficient cause for tumorigenesis remains to be determined. Consistent with our hypothesis is the fact that c-cbl-/- mice have increased kinase activity in lymphocytes, but is not sufficient for tumor formation [35], [36], [37]. c-CBL LOH could also lead to increased expression of c-CBL from the other allele to compensate for a loss of an allele. Alternately, there could be some form of synergy working with reduced c-CBL levels and mutated receptors that exacerbate the phenotype of each alone.  Previous studies from our lab and others have shown that East Asians with lung cancers have relatively high frequencies of gain-of-function of mutations in RTKs such as EGFR and MET [31]. In a cohort of Japanese patients an activating MET mutation was identified in the splice region that deletes the juxtamembrane domain that is involved E3 activity of c-CBL [38]. This study also found that activation of MET is mutually exclusive of EGFR, KRAS and HER2 gene mutations [38]. We failed to detect such mutations in significant numbers in lung tumor samples obtained from African-Americans (n\\u200a=\\u200a29) and Caucasian (n\\u200a=\\u200a50) patients. One MET mutation was identified in each of the groups whereas 1 and 3 EGFR mutations were identified in the African-American and Caucasian cohorts respectively. EGFR mutations have earlier been identified as one of the key mutations affecting lung adenocarcinoma patients in a comprehensive study of 188 patients [39]. Our study encompasses different histologies of NSCLC. However, the published series did not find any mutations in c-CBL or MET unlike our study that encompassed different subtypes of NSCLC. It is important to note, we have recently shown that MET mutations in lung cancer are in majority germline [31]. We have reported earlier c-CBL mutations in a small cohort of Taiwanese lung cancer samples [40]. In our efforts to understand the ethnic differences in the lung oncogenome, we also looked at PAX transcription factors such as PAX5 and PAX8 that are highly expressed in lung cancers; however there was no preferential expression or mutations of the above genes in lung tumor samples of African-Americans. In this study, we show relatively high frequency of c-CBL mutations in lung cancers, especially in the large cell type among Caucasians and particularly among African-Americans. We therefore propose c-CBL as an efficacious target for lung cancers in African-Americans that needs to be further substantiated. This is all the more important because the prognosis for African-Americans with lung cancer, especially for men is much poorer compared to their Caucasian counterparts [41].  In conclusion, the results presented in this study demonstrate that c-CBL is frequently mutated or even lost in lung cancers. Our results support a role for c-CBL mutants that are independent of its ubiquitination activity. Given the relatively high mutation rates in c-CBL as well as RTKs such as MET and EGFR, it is likely that their combined effect could be synergistic in promoting tumorigenesis.\""
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_text.Text[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\" Abstract Background  Non-small cell lung cancer (NSCLC) is a heterogeneous group of disorders with a number of genetic and proteomic alterations. c-CBL is an E3 ubiquitin ligase and adaptor molecule important in normal homeostasis and cancer. We determined the genetic variations of c-CBL, relationship to receptor tyrosine kinases (EGFR and MET), and functionality in NSCLC.  Methods and Findings  Using archival formalin-fixed paraffin embedded (FFPE) extracted genomic DNA, we show that c-CBL mutations occur in somatic fashion for lung cancers. c-CBL mutations were not mutually exclusive of MET or EGFR mutations; however they were independent of p53 and KRAS mutations. In normal/tumor pairwise analysis, there was significant loss of heterozygosity (LOH) for the c-CBL locus (22%, n\\u200a=\\u200a8/37) and none of these samples revealed any mutation in the remaining copy of c-CBL. The c-CBL LOH also positively correlated with EGFR and MET mutations observed in the same samples. Using select c-CBL somatic mutations such as S80N/H94Y, Q249E and W802* (obtained from Caucasian, Taiwanese and African-American samples, respectively) transfected in NSCLC cell lines, there was increased cell viability and cell motility.  Conclusions  Taking the overall mutation rate of c-CBL to be a combination as somatic missense mutation and LOH, it is clear that c-CBL is highly mutated in lung cancers and may play an essential role in lung tumorigenesis and metastasis.  Go to: Introduction In the US alone, each year approximatelypeople are diagnosed with lung cancers, out of which more thanof them succumb to the disease . This number is roughly equivalent to the combined mortality rates of cancers of the breast, prostate, colon, liver, kidney and melanoma . In addition the prognosis is usually poor and the five-year survival rate is less thanThere are also significant ethnic differences for lung cancer, and the outcome is worse for blacks compared to whites. Gender differences are also striking with women having significantly better prognosis as compared to men. There are a number of genetic alterations that can occur in lung cancer. As an example, in NSCLC, mutations in KRAS, p53, EGFR and MET have been identified. Many of these pathways, especially Receptor Tyrosine Kinases (RTKs) are controlled by c-CBL.  CBL (Casitas B-lineage lymphoma) is a mammalian gene located on human chromosome 11q23.3  and is involved in cell signaling and protein ubiquitination . CBL proteins belong to the RING finger class of ubiquitin ligases (E3) and there are three homologues c-CBL, CBL-b, CBL-3 . The c-CBL and CBL-b genes are ubiquitously expressed with the highest levels in hematopoietic tissues . c-CBL consists of four regions encoding for functionally distinct protein domains: the N-terminal tyrosine kinase binding (TKB) domain, the linker region, the catalytic RING finger domain, the proline-rich region and the c-terminal ubiquitin-associated (UBA) domain that also overlaps with a leucine-zipper (LZ) domain . Both TKB and RING finger domains are essential for ligand-induced ubiquitination of RTKs , , , . The RING finger domain is required for the recruitment of E2 ubiquitin-conjugating enzymes. The TKB domain includes four-helix bundle (4H), a calcium-biding EF hand, and a modified SH2 domain, which binds to phosphotyrosine residues , , , . In addition, the proline-rich region of c-CBL can associate with the SH3 domain of Grb2, which can indirectly recruit c-CBL to RTKs via the GRB2 adaptor protein , , .  c-CBL also binds to EGFR and acts as the E3 that targets EGFR for ubiquitination and degradation. Furthermore, CBL desensitizes EGF signaling and opposes cellular proliferation induced by EGF . EGF activation also appears to activate the tyrosine kinase SRC, which phosphorylates c-CBL and in turn activates the ubiquitination and degradation of EGFR , , . A recent study shows that defective endocytosis of EGFR is characterized by a deletion mutant and the point mutation L858R, whereby its association with c-CBL and subsequent ubiquitination are impaired . Recently, the first human c-CBL mutations were reported in acute myeloid leukemia (AML) patients . The mutation R420Q inhibits FMS-like tyrosine kinase(FLT3) internalization and ubiquitination .  Not only can E3 activity be important in oncogenesis, c-CBL has a dual but separate function as a signal transduction molecule. We have previously shown that c-CBL is important in binding CRKL and BCR/ABL in hematopoietic cells. Also, it can bind and modulate functions of cytoskeleton by binding to proteins like talin and paxillin. The TKB domain is important in binding to a number of molecules, and they then function in signal transduction.  Given the critical role of CBL in normal homeostasis and cancer, we hypothesized that it might be mutated in lung cancers. In this study, we report novel c-CBL somatic mutations S80N/H94Y, Q249E and W802* in Caucasian, Taiwanese and African-American lung cancer patients, respectively. Expressing these mutations in NSCLC cell lines lead to increased proliferation and cell motility. We show that c-CBL mutations occur with or without MET or EGFR mutations but are mutually exclusive of a LOH at the c-CBL locus. Additionally, c-CBL LOH is associated with either MET or EGFR mutations. We thus hypothesize that c-CBL mutations might contribute to the oncogenic potential of MET and EGFR in lung cancer.  Go to: Methods Ethics Statement  Written consent on all research on human subjects has been obtained from the Institutional Review Board, University of Chicago and covers all research performed in the laboratory. The following is their contact information: Institutional Review Board, The University of Chicago, McGiffert Hall,S. Woodlawn Ave., 2nd floor, Chicago, ILWritten informed consents were received from all patients whose tissue samples were used for this study.  Tissue Samples  Lung cancer tissue and paired adjacent normal lung tissues were obtained fromCaucasian,African-Americans andTaiwanese NSCLC patients who were recruited at the University of Chicago Hospital (Chicago, USA) (Caucasian and African-American patients) and Taipei Veterans General Hospital of Taiwan (Taiwanese patients) after obtaining appropriate Institutional Review Board permission and informed consent from the patients. Out ofsamples,were men,were women andwere unknown with age at diagnosis ranging fromtoyears. In terms of tumor types,were adenocarcinoma,were squamous cell carcinoma andwere large cell carcinoma.were stage I,were stage II,were stage III, andwere stage IV (Table S1).  Cell Culture  Human non-small cell lung carcinoma cells A549 and H358 were maintained in DMEM and RPMI-1640, respectively. Human embryonic kidney 293T cells were cultured in DMEM. Media were supplemented withfetal bovine serum,units/ml of penicillin, andµg/ml of streptomycin (Invitrogen, Carlsbad, CA). Cells were cultured at 37°C in a humidified incubator containingCO2.  c-CBL Gene Mutational Analysis  Exonstoof c-CBL gene were individually amplified by polymerase chain reaction (PCR). Primers are listed in Table S2. PCR conditions werecycle of 95°C forminutes;cycles of 94°C forseconds, 58°C forseconds and 72°C forminutes; and one cycle of 72°C forminutes. PCR products were treated with ExoSAP-IT (USB Corporation, Cleveland, OH) and sequenced by Big-Dye Terminator Chemistry (Applied Biosystems, Foster City, CA). Sequencing was performed on the forward coding strand with confirmation of c-CBL alterations performed by sequencing the reverse strand as well. Chromatograms were analyzed for mutations using Mutation Surveyor v2.61 (Softgenetics, State College, PA).  Plasmid Constructs and Site-Directed Mutagenesis  The wild-type c-CBL cDNA insert was subcloned into the pAlterMax expression vector using XhoI and SalI restriction enzyme sites (Promega, Madison, WI). Using this parental plasmid pAlterMax-c-CBL, the TKB domain double mutation (S80N/H94Y), the point mutation (Q249E), and the C-terminal point mutation W802* of c-CBL were created using the following primers: 5′-GCTGGCGCTAAAGAATAACCCACCTTATATCTTAGAC-3′ and 5′-CTACCAGATACCTACCAGTATCTCCGTACTATCTTGTC-3′ for the double mutation S80N/H94Y; 5′-CTTTACCCGACTCTTTGAGCCCTGGTCCTCTTTGC-3′ for Q249E, and 5′-CAGCTCCTCCTTTGGCTGATTGTCTCTGGATGGTGATC-3′ for W802* along with their complementary primers using the QuickChange Site-Directed Mutagenesis XL kit (Stratagene, La Jolla, CA) according to the manufacturer's instructions. The constructs were confirmed for the point mutations by standard DNA sequencing of both strands.  Loss of Heterozygosity (LOH) Analysis  Five microsatellites on chromosome(3 on 11q at or withinkb up or downstream of the c-CBL gene andcontrol markers on 11p) were selected for analysis (Table S3). Established microsatellite markers and respective primer sequences were selected from the GeneLoc database (http://genecards.weizmann.ac.il/geneloc/index.shtml, Weizmann Institute of Science, Rehovot, Israel). Primers were custom designed and each forward primer was fluorescently labeled at the 5′ end with FAM, PET, NED, or VIC (Applied Biosystems). Primer annealing temperatures and duplex scores were evaluated with NIST Primer Tools (http://yellow.nist.gov:8444/dnaAnalysis/primerToolsPage.do; National Institute of Standards and Technology, Gaithersburg, MD). Primers were verified by performing PCR with control DNA (isolated from TK6 cells) and resolving the products on agarose gels. Bands were visualized with an UV transilluminator. Genomic DNA was extracted from tumor samples and paired normal lung tissue. Primers were grouped into multiplex combinations shown in Table S4. Marker D11S929 served as an internal control to check for consistency in PCRs and of peaks from capillary electrophoresis. Multiplex PCRs were carried out in a volume ofµL that containedµL genomic DNA (20–50 ng),µM of each primer (1.0 µM total for each primer pair),µM dNTPs, 1X PCR buffer containing MgCl2, andU Taq DNA polymerase. PCR was performed on the ABI GeneAmpPCR System under the following conditions:min at 94°C;cycles ofsec at 94°C,min at 60°C,min at 72°C; andmin at 72°C. The PCR products were separated by capillary electrophoresis on an ABI 3130XL DNA Analyzer. Chromatograms were analyzed with Peak Scannerand GeneMappersoftware (Applied Biosystems) for allelic alterations. The area of the peaks produced by the DNA PCR products was quantified for each allele. The ratio of the allelic areas was calculated for each tumor and paired normal DNA sample. When the qLOH (allelic ratio for the tumor peaks divided by the allelic ratio of paired normal sample) was ≤0.5 or ≥2.0 for c-CBL and at least one other 11q marker in at least two separate experiments, the sample was considered as having an allelic imbalance and interpreted as LOH. Samples were evaluated in at least two separate experiments and samples showing prospective LOH at c-CBL repeated a third time which included a new control marker at the BAX locus (data not shown) on chromosometo verify integrity of sample DNA.  Transfection of c-Cbl Constructs  The A549 cell line was transfected using the Fugene HD (Roche, Nutley, NJ) reagent according to the manufacturer's instructions. Eight µg of plasmid DNA, containing either no insert (empty vector), wild-type c-CBL, S80N/H94Y c-CBL, Q249E c-CBL or W802* CBL was used for transfection in a 6-well culture plate. Cells were harvestedh after transfection and analyzed for expression.  c-CBL Knockdown  c-CBL knockdown was performed using lentiviral transduction using MISSION lentiviral transduction particles (Sigma-Aldrich, St. Louis, MO) as per manufacturer's instructions. Briefly, 1×105 H358 cells/well were seeded in 6-well plates and infected the following day with c-CBL lentiviral shRNA constructs. To generate stable c-CBL knockdown cell lines, cells were selected fordays withµg/ml puromycin. c-CBL levels were determined using whole cell lysates by immunoblotting with anti-CBL antibody (Santa Cruz Biotechnologies, Santa Cruz, CA).  Cell Viability Assay  Cells were transfected as described above in the transfection assay. Forty-eight hours after transfection, viability of cells was assessed using Trypan Blue exclusion.  Wound Healing Assay  A549 cells were seeded in 6-well plates and cultured forh untilconfluent. The medium was then changed and the cells were transfected as described in the transfection assay. Twelve hours after transfection, a straight scratch was made across the cell layer using aml pipette tip. The cells were then gently washed with 1× PBS to remove cellular debris and the media was replaced. Photographs were taken of the wound region everyh untilh.  Western Blot Analysis  Forty eight hours after transfection, cells were collected and washed twice in 1X PBS, then lysed in ice-cold lysis buffer (0.5M Tris-HCl with pH1.5 M NaCl,deoxycholic acid,mM EDTA,NP-40,mM DTT,mM phenylmethylsulfonyl fluoride,µg/mL leupeptin, andµg/mL aprotinin) forminutes. The lysate was centrifuged atrpm forminutes at 4°C, and protein content of the supernatant was measured. Total cell lysates (50 µg/well) were separated by SDS-PAGE electrophoresis and the gels transferred onto nitrocellulose membranes (Whatman, Piscataway, NJ). Membranes were blocked withnon-fat dry milk in phosphate-buffered saline containing Tween-20 (PBST) (1X PBS,Tween-20) forh at room temperature and incubated with the appropriate primary antibody at 4°C overnight. Membranes then were washed three times with PBST and probed with appropriate horseradish peroxidase (HRP)-conjugated secondary antibody forh at room temperature. The membranes were again washed three times in PBST and bands were visualized using Western blot chemiluminescence reagent (BioRad, Valencia, CA) on a Chemidoc Gel documentation system (BioRad, Valencia, CA). Antibodies were obtained from Santa Cruz Biotechnologies and used at the following dilutions (c-CBL, 1∶5000; c-MET, 1∶5000; EGFR, 1∶5000; ubiquitin, 1∶1000; HA, 1∶5000 and β-actin, 1∶10,000).  Flow Cytometry  Cell cycle analysis was carried out by flow cytometry. Approximately 2×106 cells were grown in media containingFBS. Cells were harvested by trypsin/EDTA treatment, washed with 1X PBS three times and fixed with ice-coldethanol forh. Cells were washed again with cold PBS and stained with a solution containingµg/mL propidium iodide,µg/mL RNase A, andTriton X-100 forminutes in the dark. Cell cycle analysis was performed using a Guava PCA-96 flow cytometer (Guava Technologies, Millipore, Billerica, MA).  Ubiquitin Ligase Activity  293T cells were maintained in culture in DMEM supplemented withFBS andpenicillin (100 units/mL) and streptomycin (100 µg/mL) were transfected withµg EGFR-pcDNA3 andµg HA-tagged c-CBL constructs as indicated using calcium phosphate according to manufacturer's protocol (Profection, Promega, Madison, WI). Twenty-four hours post-transfection, cells were starved overnight in DMEM supplemented withFBS, and then treated with or without EGF (100 ng/ml) formin. The cells were collected and washed two times in ice-cold PBS containingmM sodium orthovanadate then lysed in ice-cold lysis buffer (10 mM Tris HCl, pH150 mM NaCl,mM EDTA,Triton X100,Glycerol,mM sodium orthovanadate and protease inhibitors). Lysates were cleared of debris by centrifugation atg formin at 4°C. EGFR immunoprecipitations were performed onµg of cleared lysate usingng of rabbit-anti-EGFR and Protein A/G Plus Sepharose overnight at 4°C. Precipitations were washedtimes in lysis buffer before boiling in Laemmli buffer. Elutions were immunoblotted with anti-ubiquitin and EGFR. Twenty micrograms of cleared lysate were immunoblotted for each of the c-CBL constructs using anti-HA.  Statistical Analysis  Mutation rates between different groups were compared using Fisher's exact test. For continuous variables, group comparisons were performed using analysis of variance (ANOVA) followed by Sidak's adjustment for multiple comparisons. Experiments involving repeated measurements over time were analyzed using repeated measures ANOVA with the Greenhouse-Geisser adjustment to the degrees of freedom. Analyses were conducted using STATA (v10.1) software (Stata Corporation, College Station, TX).  Go to: Results c-CBL Gene Mutations in Lung Cancer  To investigate the role of c-CBL in lung cancer, we analyzed its genomic DNA in tumor and paired normal samples drawn from multiple ethnicities. The lung tumor samples represented Caucasians (n\\u200a=\\u200a50), African-Americans (n\\u200a=\\u200a29), and Taiwanese (n\\u200a=\\u200a40) lung cancer patients. We designedpairs of primers to sequence the coding region of c-CBL gene that spans exonsto(Table S2). We identifiedunique somatic mutations in c-CBL exons amongdifferent patients. A variation L620F, a known SNP (rs2227988) in exonwas also detected. Importantly, the eight novel non-synonymous mutations were confirmed by sequencing both strands of c-CBL genomic DNA obtained from lung tumor samples (Table 1). Moreover, none of themutations were detected in the corresponding normal tissue, indicating that these were somatic mutations. Four synonymous single nucleotide variations (SNVs) were also identified but were not used further in this study.  TableTablec-CBL mutation analysis inlung cancer patient tumor tissues. Three of thenovel non-synonymous mutations were located in the TKB (tyrosine kinase binding) domain (S80N, H94Y, and Q249E), one in the RING finger domain (V391I), one in the proline-rich region (72515_72517 del ATG), and three in the C-terminal region (W802*, R830K, and A848T) of the c-CBL protein ( and ). In , we show model chromatograms of representative samples.  igurec-CBL mutations and LOH in non-small cell lung cancer. 11q LOH of c-CBL Gene  Paired lung tumor and normal lung tissue samples from Taiwanese patients (n\\u200a=\\u200a37) were investigated for LOH. Eight (21.6%) showed LOH at the c-CBL locus on chromosomewhilesamples (78.4%) revealed normal allelic contribution at the microsatellite markers (Figures 1C and D).  c-CBL Mutations in Different Ethnic Groups  The c-CBL double mutant S80N/H94Y was found in the same patient and the overall mutation rate for c-CBL in lung tumors was(8/119). The frequency of c-CBL mutation was highest in large cell carcinoma (14.7%;ofpatients) followed by squamous carcinoma (6.3%;ofpatients) and the least was observed in adenocarcinoma (AD) (1.8%;ofpatients), although these rates were not statistically significant (p\\u200a=\\u200a0.292). Mutation rates wereamong Caucasians (0 ofin AD;ofin SQ; andofin LC),in African-Americans (1 ofin AD;ofin SQ; andofin LC), and(0 ofin AD;ofin SQ; andofin LC) in the Taiwanese population. Additionally two Taiwanese patients with lung cancer (one squamous and one adenocarcinoma) had the known SNP L620F. Ethnic differences were not statistically significant, however the power to detect differences was low.  Mutations in MET and EGFR Can Be Co-Associated with c-CBL Alterations  Since East Asians with lung cancer have a higher frequency of EGFR and MET mutations in lung tumors , , we also determined mutations in EGFR and MET in the same Taiwanese cohort samples and compared the results with the observed c-CBL alterations (LOH and/or mutations). In thesamples tested, we did not find any overlap between c-CBL mutations and c-CBL LOH . Of the three c-CBL mutants (including the known L620F SNP, rs2227988), one of the samples had a MET mutation (N375S) and the other had an EGFR mutation (L858R). Among thesamples that had a LOH at the c-CBL locushad an additional mutation in MET (N375S) andhad an EGFR exondeletion. Twenty-six samples had neither c-CBL mutation nor c-CBL LOH (3 patients had a c-CBL mutation but no c-CBL LOH). Among thesesampleshad a MET mutation (8 N375S,L211W),had an EGFR mutation (7 exondeletion,L858R) andhad no other MET or EGFR mutation. Thus the rate of MET or EGFR mutations among patients with LOH at the c-CBL locus (7 of 8) was similar to that seen in patients without c-CBL mutation or LOH (22 ofpatients) (p\\u200a=\\u200a0.99). Thesepatients with no identifiable mutation in c-CBL, MET or EGFR representedof thepatients analyzed in the Taiwanese patient cohort. Conversely,Taiwanese lung cancer patients have an identifiable mutation in either c-CBL, MET or EGFR or a combination of the three genes . Additionally, we determined p53 and KRAS mutations in these Taiwanese cohorts. Two p53 andKRAS mutation were detected. The single KRAS mutation overlapped with one p53 mutation. This patient also had the EGFR exondeletion but had no c-CBL mutation. The other p53 mutation sample had a c-CBL LOH with concurrent MET N375S mutation. Thus, in the Taiwanese samples analyzed, p53/KRAS mutations and c-CBL mutations were mutually exclusive (data not shown).  igurec-CBL mutations and relationship to MET and EGFR mutations in lung cancer. Cellular Functions of c-CBL Alterations in the Context of Lung Tumorigenesis  A. E3 activity is intact in the mutant c-CBL proteins To investigate whether the different c-CBL mutations affect the E3 activity, EGFR was chosen as a model substrate for c-CBL E3 function. All of the c-CBL mutants tested enhanced ubiquitination of the activated EGFR similar to the wild-type c-CBL protein. This result demonstrates that the catalytic activity of the c-CBL mutants is not impaired when EGFR was the substrate. .  igureUbiquitination, viability, expression and cell cycle analysis of various c-CBL mutants. B. Effect on lung cancer cell viability The effect of a representative c-CBL mutant from each of the three ethnic backgrounds on lung cancer cell viability in cell lines was determined. S80N/H94Y double mutation, Q249E, and W802* were identified in lung tumor samples obtained from a Caucasian, a Taiwanese and an African-American, respectively. As described in methods, the c-CBL wild-type (WT) and the above three mutants were expressed after cloning them into pAlterMax vector in A549 cells. These cells express relatively low basal levels of endogenous c-CBL (data not shown). Transfection efficiency was comparable between different groups and the number of cells transfected with c-CBL wild-type construct was aboutcompared to control cells that were transfected with the empty vector. Cells transfected with S80N/H94Y, Q249E and W802* c-CBL mutant constructs resulted in increased number of viable cells that was120.8% andhigher respectively, relative to the empty vector control transfected cells and significantly different from the wild-type construct (p\\u200a=p\\u200a=and p\\u200a=respectively) . Relative levels of c-CBL protein in whole cell lysates prepared from samples obtained from a parallel experiment was determined. The c-CBL protein levels in samples representing untransfected and empty vector transfected cells were comparable and those representing the c-CBL WT and the three c-CBL mutants were comparable .  C. Effect on cell cycle To investigate if the increases in cell viability in different c-CBL mutants are due to increased cellular proliferation, a cell cycle analysis was performed. A549 cells were transfected with the c-CBL WT or the three different mutants: S80N/H94Y, Q249E and W802*. The empty vector transfectant was used as a control. Forty-eight hours after transfection cell cycle analysis was performed as described in materials and methods. There was no significant change in the subG1, G1 or the S phase of the cell cycle among the different mutants compared to the WT construct (p\\u200a=p\\u200a=and p\\u200a=respectively). The G2/M phase of the cell cycle showed an increase in cell numbers for the three mutants, S80N/H94Y, Q249E and W802*, when compared to the WT but again the difference was not statistically significant (p\\u200a=\\u200a0.25) .  D. Effect on cell motility To investigate the effect of the expression of the above three c-CBL mutants on cell migration, we carried out wound healing assay as described in materials and methods. The closing of the scratch or the wound was monitored at12,36, andh. . In all the samples, that represented cells transfected with mutants, the wound gap was much smaller than that seen in the sample that represented cells transfected with c-CBL WT (p<0.001). We also determined the rate of wound closure for all the five groups. Ath, wild-type c-CBL transfectants showedopen wound while the S80N/H94Y, Q249E and W802* mutants showed23.9% andopen wound respectively (p<0.001) .  igurec-CBL mutations affect wound healing in A549 cells. E. c-CBL knockdown increases cell viability It is hypothesized that the LOH seen in our samples could lead to decreased expression of c-CBL. Thus we tested the effect of c-CBL knockdown in lung cancer cells. Compared to A549, H358 lung cancer cells express relatively high levels of endogenous c-CBL (data not shown). c-CBL expression was knocked down using lentiviral construct that expressed c-CBL specific shRNA and compared the results with those that were transduced with scrambled shRNA and the results are shown in e identified several clones that revealed varying degrees of c-CBL knockdown showing different sets of c-CBL lentiviral shRNA knockdown efficiency . Of all the clones tested, Clonewas chosen for further experiments. Equal amount of cells were seeded in a 6-well plate and the cell proliferation was measured at various times and the results are depicted in . As expected, number of cells increased in a time dependent fashion fromtorelative to scrambled shRNA as control in a span ofh (p\\u200a=\\u200a0.0002) . The cell cycle phases in H358 cells that were knocked down with c-CBL shRNA were looked at and compared with the scrambled shRNA. There were no discernable differences between these two constructs in the different phases of the cell cycle (data not shown).  igureKnockdown of c-CBL using an shRNA increases cell proliferation. Go to: Discussion Our results demonstrate that c-CBL is somatically mutated (or has LOH) in lung cancers, and can significantly contribute to enhanced cell viability and motility. There was also a high prevalence of LOH with respect to c-CBL in lung tumors that harbored MET or EGFR mutation.  In the present study, we have demonstrated the occurrence of c-CBL mutations in lung cancer patients, especially with different ancestral variations. Mutations in c-CBL have been recently reported in juvenile myelomonocytic leukemia and myeloid malignancies. In the AML study, the mutation R420Q located in the junction of the RING finger and the linker region inhibited FMS-like tyrosine kinase(FLT3) internalization and ubiquitination , thus contributing to the gain-in-function for the RTK. In addition, mutations such as H398Y, C384R, and L380P mapped to the RING finger domain and the linker region of c-CBL that is required for its E3 activity , , , , . Additionally, homozygous mutations in the RING finger domain of the c-CBL gene were described as a result of acquired Uniparental Disomy (UPD) . It is important to note that our results indicate LOH at 11q23 locus and these are mutually exclusive from missense mutations of c-CBL. The somatic mutations were all heterozygous. The mutations in AML led to abrogation of the E3 activity leading to prolonged RTK activation. In addition mutants located on the linker region surrounding the RING finger domain exhibited enhanced AKT signaling in response to cytokine stimulation . In addition, it was shown in NH3T3 cells, that neither mutations in the RING finger nor the linker region causes transformation, however while certain mutations perturbs the ubiquitination, others affect receptor recycling and prolong kinase activity .  We report here c-CBL mutations that mapped not only to the RING finger domain, but also to the TKB domain, proline-rich domain and the C-terminal region, but none mapped to the linker region as reported in the AML studies described above , , , , . In addition,mutants that we detected were found in different ethnic backgrounds. For example, S80N/H94Y, Q249E, W802* were detected in Caucasians, Taiwanese and African-Americans, respectively. The results point out not only the difference between lung cancer and other cancers, but also genetic polymorphism among different races in the same cancer. Interestingly, there is a large disparity between African-American and other ethnic populations with lung cancer . We have previously shown that there was a low frequency of EGFR and MET mutation in African-Americans as compared with Taiwanese and Caucasians . In this study, the number of African-American samples analyzed was relatively fewer and we foundmutations that are unique to this ethnicity. It would now behoove us to further study the genetic alterations that can occur and determine the targeted therapeutics for African-Americans. Our results provide evidence of the importance of c-CBL in tumorigenesis and potential signaling. Our prediction, based on the AML data, would be that the V391I RING finger domain mutation would affect the E3 activity. Also, it will be important to determine the binding partners for c-CBL in the TKB domain and proline rich domain mutations. It has previously been shown that the TKB domain can bind to growth factor receptors and it will be important to determine the cross-binding of these mutants to MET and or EGFR. It would also be important in the future to look at fluorescence in situ hybridization/copy number changes in c-CBL in lung cancer.  c-CBL plays an important role in down regulating RTK-mediated signaling through K63 poly-ubiquitination and subsequent downregulation of RTKs followed by lysosomal degredation . Mono-ubiquitination or ubiquitinated with K63-linked chains of substrates by c-CBL may lead to enhancement of biological and biochemical functions (reviewed in Hermann et al,). The mutations that we analyzed in our studies all point out to the fact that the E3 activity of c-CBL on EGFR is intact; the EGFR levels in the various mutants remain same . Multiple kinases, both RTKs and non-RTKs could be acted upon by c-CBL, including ERBs, PDGFR, FMS, MET, c-Kit, VEGFR, FLT-1, RON, FGFR, IR, as well as SYK, FYN, LCK, FGR, LYN and c-ABL . In lung cancers the relevant substrates of c-CBL in terms of degradation or signal transduction are yet to be identified.  The observation that c-CBL somatic mutations, especially S80N/H94Y, Q249E and W802* showed increased cell viability and cell motility that is in agreement with the physiological role for D-cbl in the regulation of apoptosis and differentiation identified in Drosophila is very significant . It has been previously shown that activating c-CBL mutation downregulates EGFR signaling and decreases cellular proliferation and migration in breast cancer cell lines . Although the role of c-CBL in the negative regulation of RTKs is well substantiated, thereby suggesting that it is a natural tumor suppressor, studies in cancer cells have revealed both tumor suppressor and tumor promoting activities depending on the type of c-CBL mutation and the number of alleles at the c-CBL locus . In agreement with the above, the three c-CBL mutants described here appear to have tumor growth and metastasis promoting properties. Although these mutants are outside of the RING finger or the linker region of c-CBL, their downstream effects are significant so as to cause increased proliferation and migration, but the substrate affected by these mutations are not known yet. This raises the possibility that some of the cellular functions of c-CBL are independent of its ubiquitin-ligase activity, an area that we are currently investigating. The oncogenic nature of RTKs, addiction of cancers to growth signals and given the clustering of c-CBL, EGFR and MET mutations, it is possible that the transforming effect of c-CBL mutations is most likely a combinatorial effect of the three. We also show that LOH for c-CBL was found in a significant number of samples that harbored MET or EGFR mutations. The fact that aboutof lung tumor samples are likely to have c-CBL mutations and an additionalare likely to harbor c-CBL-related LOH, makes c-CBL a highly mutated molecule in lung cancer. Since LOH alone is not enough to cause a transforming event , , , associated mutation in the MET or EGFR locus or yet another RTK discussed above may play a role in carcinogenesis. We predict that this LOH in c-CBL results in haploinsufficiency that downplays RTK ubiquitination leading to hyperactivity of the RTKs. However, whether this is sufficient cause for tumorigenesis remains to be determined. Consistent with our hypothesis is the fact that c-cbl-/- mice have increased kinase activity in lymphocytes, but is not sufficient for tumor formation , , . c-CBL LOH could also lead to increased expression of c-CBL from the other allele to compensate for a loss of an allele. Alternately, there could be some form of synergy working with reduced c-CBL levels and mutated receptors that exacerbate the phenotype of each alone.  Previous studies from our lab and others have shown that East Asians with lung cancers have relatively high frequencies of gain-of-function of mutations in RTKs such as EGFR and MET . In a cohort of Japanese patients an activating MET mutation was identified in the splice region that deletes the juxtamembrane domain that is involved E3 activity of c-CBL . This study also found that activation of MET is mutually exclusive of EGFR, KRAS and HER2 gene mutations . We failed to detect such mutations in significant numbers in lung tumor samples obtained from African-Americans (n\\u200a=\\u200a29) and Caucasian (n\\u200a=\\u200a50) patients. One MET mutation was identified in each of the groups whereasandEGFR mutations were identified in the African-American and Caucasian cohorts respectively. EGFR mutations have earlier been identified as one of the key mutations affecting lung adenocarcinoma patients in a comprehensive study ofpatients . Our study encompasses different histologies of NSCLC. However, the published series did not find any mutations in c-CBL or MET unlike our study that encompassed different subtypes of NSCLC. It is important to note, we have recently shown that MET mutations in lung cancer are in majority germline . We have reported earlier c-CBL mutations in a small cohort of Taiwanese lung cancer samples . In our efforts to understand the ethnic differences in the lung oncogenome, we also looked at PAX transcription factors such as PAX5 and PAX8 that are highly expressed in lung cancers; however there was no preferential expression or mutations of the above genes in lung tumor samples of African-Americans. In this study, we show relatively high frequency of c-CBL mutations in lung cancers, especially in the large cell type among Caucasians and particularly among African-Americans. We therefore propose c-CBL as an efficacious target for lung cancers in African-Americans that needs to be further substantiated. This is all the more important because the prognosis for African-Americans with lung cancer, especially for men is much poorer compared to their Caucasian counterparts .  In conclusion, the results presented in this study demonstrate that c-CBL is frequently mutated or even lost in lung cancers. Our results support a role for c-CBL mutants that are independent of its ubiquitination activity. Given the relatively high mutation rates in c-CBL as well as RTKs such as MET and EGFR, it is likely that their combined effect could be synergistic in promoting tumorigenesis.\""
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "process_text(df_text.Text[1])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now that we have a processor that can clean up the text, let us clean it up all at once"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df_processed = df_text.Text.map(process_text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    Cyclin-dependent kinases (CDKs) regulate a var...\n",
       "1     Abstract Background  Non-small cell lung canc...\n",
       "2     Abstract Background  Non-small cell lung canc...\n",
       "3    Recent evidence has demonstrated that acquired...\n",
       "4    Oncogenic mutations in the monomeric Casitas B...\n",
       "Name: Text, dtype: object"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_processed.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### TFIDF\n",
    "\n",
    "Now that we have cleaned up the text data, we should try to run a tfidf on it.  However, I will first use a Porter tokenizer which will map the words running, runs, ran, etc to the same stem 'ru'.  This allows us to ignore the tenses of words, and just concentrate on the syntactic meaning of them.  After we vectorize paragraphs using this, we can run a simple logistic regression on the vectors.   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from sklearn.feature_extraction.text import TfidfVectorizer\n",
    "from sklearn.pipeline import Pipeline\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.model_selection import cross_val_score\n",
    "import nltk\n",
    "from nltk.stem.porter import PorterStemmer\n",
    "def port_tokenizer(text):\n",
    "    words = re.split(r'[^a-z0-9-]+',text.lower())\n",
    "    ps = PorterStemmer()\n",
    "    return [ps.stem(word) for word in words if not word in nltk.corpus.stopwords.words('english') and word]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To make classification easier, we will first only try to classify texts as '7' and 'not-7'.    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "y = df_variants.Class.map(lambda x: x==7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['cyclin-depend',\n",
       " 'kinas',\n",
       " 'cdk',\n",
       " 'regul',\n",
       " 'varieti',\n",
       " 'fundament',\n",
       " 'cellular',\n",
       " 'process',\n",
       " 'cdk10',\n",
       " 'stand',\n",
       " 'one',\n",
       " 'last',\n",
       " 'orphan',\n",
       " 'cdk',\n",
       " 'activ',\n",
       " 'cyclin',\n",
       " 'identifi',\n",
       " 'kinas',\n",
       " 'activ',\n",
       " 'reveal',\n",
       " 'previou',\n",
       " 'work',\n",
       " 'shown',\n",
       " 'cdk10',\n",
       " 'silenc',\n",
       " 'increas',\n",
       " 'ets2',\n",
       " 'v-et',\n",
       " 'erythroblastosi',\n",
       " 'viru',\n",
       " 'e26',\n",
       " 'oncogen',\n",
       " 'homolog',\n",
       " '2',\n",
       " '-driven',\n",
       " 'activ',\n",
       " 'mapk',\n",
       " 'pathway',\n",
       " 'confer',\n",
       " 'tamoxifen',\n",
       " 'resist',\n",
       " 'breast',\n",
       " 'cancer',\n",
       " 'cell',\n",
       " 'precis',\n",
       " 'mechan',\n",
       " 'cdk10',\n",
       " 'modul',\n",
       " 'ets2',\n",
       " 'activ',\n",
       " 'gener',\n",
       " 'function',\n",
       " 'cdk10',\n",
       " 'remain',\n",
       " 'elus',\n",
       " 'demonstr',\n",
       " 'cdk10',\n",
       " 'cyclin-depend',\n",
       " 'kinas',\n",
       " 'identifi',\n",
       " 'cyclin',\n",
       " 'activ',\n",
       " 'cyclin',\n",
       " 'cyclin',\n",
       " 'orphan',\n",
       " 'cyclin',\n",
       " 'product',\n",
       " 'fam58a',\n",
       " 'whose',\n",
       " 'mutat',\n",
       " 'caus',\n",
       " 'star',\n",
       " 'syndrom',\n",
       " 'human',\n",
       " 'development',\n",
       " 'anomali',\n",
       " 'whose',\n",
       " 'featur',\n",
       " 'includ',\n",
       " 'toe',\n",
       " 'syndactyli',\n",
       " 'telecanthu',\n",
       " 'anogenit',\n",
       " 'renal',\n",
       " 'malform',\n",
       " 'show',\n",
       " 'star',\n",
       " 'syndrome-associ',\n",
       " 'cyclin',\n",
       " 'mutant',\n",
       " 'unabl',\n",
       " 'interact',\n",
       " 'cdk10',\n",
       " 'cyclin',\n",
       " 'silenc',\n",
       " 'phenocopi',\n",
       " 'cdk10',\n",
       " 'silenc',\n",
       " 'increas',\n",
       " 'c-raf',\n",
       " 'confer',\n",
       " 'tamoxifen',\n",
       " 'resist',\n",
       " 'breast',\n",
       " 'cancer',\n",
       " 'cell',\n",
       " 'cdk10',\n",
       " 'cyclin',\n",
       " 'phosphoryl',\n",
       " 'ets2',\n",
       " 'vitro',\n",
       " 'cell',\n",
       " 'posit',\n",
       " 'control',\n",
       " 'ets2',\n",
       " 'degrad',\n",
       " 'proteasom',\n",
       " 'ets2',\n",
       " 'protein',\n",
       " 'level',\n",
       " 'increas',\n",
       " 'cell',\n",
       " 'deriv',\n",
       " 'star',\n",
       " 'patient',\n",
       " 'increas',\n",
       " 'attribut',\n",
       " 'decreas',\n",
       " 'cyclin',\n",
       " 'level',\n",
       " 'altogeth',\n",
       " 'result',\n",
       " 'reveal',\n",
       " 'addit',\n",
       " 'regulatori',\n",
       " 'mechan',\n",
       " 'ets2',\n",
       " 'play',\n",
       " 'key',\n",
       " 'role',\n",
       " 'cancer',\n",
       " 'develop',\n",
       " 'also',\n",
       " 'shed',\n",
       " 'light',\n",
       " 'molecular',\n",
       " 'mechan',\n",
       " 'underli',\n",
       " 'star',\n",
       " 'syndrom',\n",
       " 'cyclin-depend',\n",
       " 'kinas',\n",
       " 'cdk',\n",
       " 'play',\n",
       " 'pivot',\n",
       " 'role',\n",
       " 'control',\n",
       " 'number',\n",
       " 'fundament',\n",
       " 'cellular',\n",
       " 'process',\n",
       " 'human',\n",
       " 'genom',\n",
       " 'containsgen',\n",
       " 'encod',\n",
       " 'protein',\n",
       " 'consid',\n",
       " 'member',\n",
       " 'cdk',\n",
       " 'famili',\n",
       " 'owe',\n",
       " 'sequenc',\n",
       " 'similar',\n",
       " 'bona',\n",
       " 'fide',\n",
       " 'cdk',\n",
       " 'known',\n",
       " 'activ',\n",
       " 'cyclin',\n",
       " 'although',\n",
       " 'discov',\n",
       " 'almosti',\n",
       " 'ago',\n",
       " 'cdk10',\n",
       " 'remain',\n",
       " 'one',\n",
       " 'two',\n",
       " 'cdk',\n",
       " 'without',\n",
       " 'identifi',\n",
       " 'cyclin',\n",
       " 'partner',\n",
       " 'knowledg',\n",
       " 'gap',\n",
       " 'larg',\n",
       " 'imped',\n",
       " 'explor',\n",
       " 'biolog',\n",
       " 'function',\n",
       " 'cdk10',\n",
       " 'act',\n",
       " 'posit',\n",
       " 'cell',\n",
       " 'cycl',\n",
       " 'regul',\n",
       " 'cell',\n",
       " 'tumor',\n",
       " 'suppressor',\n",
       " 'other',\n",
       " 'cdk10',\n",
       " 'interact',\n",
       " 'ets2',\n",
       " 'v-et',\n",
       " 'erythroblastosi',\n",
       " 'viru',\n",
       " 'e26',\n",
       " 'oncogen',\n",
       " 'homolog',\n",
       " '2',\n",
       " 'transcript',\n",
       " 'factor',\n",
       " 'inhibit',\n",
       " 'transcript',\n",
       " 'activ',\n",
       " 'unknown',\n",
       " 'mechan',\n",
       " 'cdk10',\n",
       " 'knockdown',\n",
       " 'derepress',\n",
       " 'ets2',\n",
       " 'increas',\n",
       " 'express',\n",
       " 'c-raf',\n",
       " 'protein',\n",
       " 'kinas',\n",
       " 'activ',\n",
       " 'mapk',\n",
       " 'pathway',\n",
       " 'induc',\n",
       " 'resist',\n",
       " 'mcf7',\n",
       " 'cell',\n",
       " 'tamoxifen',\n",
       " 'deorphan',\n",
       " 'cdk10',\n",
       " 'identifi',\n",
       " 'cyclin',\n",
       " 'product',\n",
       " 'fam58a',\n",
       " 'bind',\n",
       " 'partner',\n",
       " 'mutat',\n",
       " 'gene',\n",
       " 'predict',\n",
       " 'absenc',\n",
       " 'truncat',\n",
       " 'cyclin',\n",
       " 'associ',\n",
       " 'star',\n",
       " 'syndrom',\n",
       " 'whose',\n",
       " 'featur',\n",
       " 'includ',\n",
       " 'toe',\n",
       " 'syndactyli',\n",
       " 'telecanthu',\n",
       " 'anogenit',\n",
       " 'renal',\n",
       " 'malform',\n",
       " 'heterozyg',\n",
       " 'femal',\n",
       " 'howev',\n",
       " 'function',\n",
       " 'cyclin',\n",
       " 'pathogenesi',\n",
       " 'star',\n",
       " 'syndrom',\n",
       " 'remain',\n",
       " 'unknown',\n",
       " 'show',\n",
       " 'recombin',\n",
       " 'cdk10',\n",
       " 'cyclin',\n",
       " 'heterodim',\n",
       " 'activ',\n",
       " 'protein',\n",
       " 'kinas',\n",
       " 'phosphoryl',\n",
       " 'ets2',\n",
       " 'vitro',\n",
       " 'cyclin',\n",
       " 'silenc',\n",
       " 'phenocopi',\n",
       " 'cdk10',\n",
       " 'silenc',\n",
       " 'increas',\n",
       " 'c-raf',\n",
       " 'phospho-erk',\n",
       " 'express',\n",
       " 'level',\n",
       " 'induc',\n",
       " 'tamoxifen',\n",
       " 'resist',\n",
       " 'estrogen',\n",
       " 'receptor',\n",
       " 'er',\n",
       " 'breast',\n",
       " 'cancer',\n",
       " 'cell',\n",
       " 'show',\n",
       " 'cdk10',\n",
       " 'cyclin',\n",
       " 'posit',\n",
       " 'control',\n",
       " 'ets2',\n",
       " 'degrad',\n",
       " 'proteasom',\n",
       " 'phosphoryl',\n",
       " 'two',\n",
       " 'neighbor',\n",
       " 'serin',\n",
       " 'final',\n",
       " 'detect',\n",
       " 'increas',\n",
       " 'ets2',\n",
       " 'express',\n",
       " 'level',\n",
       " 'cell',\n",
       " 'deriv',\n",
       " 'star',\n",
       " 'patient',\n",
       " 'demonstr',\n",
       " 'attribut',\n",
       " 'decreas',\n",
       " 'cyclin',\n",
       " 'express',\n",
       " 'level',\n",
       " 'observ',\n",
       " 'cell',\n",
       " 'previou',\n",
       " 'sectionnext',\n",
       " 'sectionresultsa',\n",
       " 'yeast',\n",
       " 'two-hybrid',\n",
       " 'y2h',\n",
       " 'screen',\n",
       " 'unveil',\n",
       " 'interact',\n",
       " 'signal',\n",
       " 'cdk10',\n",
       " 'mous',\n",
       " 'protein',\n",
       " 'whose',\n",
       " 'c-termin',\n",
       " 'half',\n",
       " 'present',\n",
       " 'strong',\n",
       " 'sequenc',\n",
       " 'homolog',\n",
       " 'human',\n",
       " 'fam58a',\n",
       " 'gene',\n",
       " 'product',\n",
       " 'whose',\n",
       " 'propos',\n",
       " 'name',\n",
       " 'cyclin',\n",
       " 'thu',\n",
       " 'perform',\n",
       " 'y2h',\n",
       " 'mate',\n",
       " 'assay',\n",
       " 'determin',\n",
       " 'whether',\n",
       " 'human',\n",
       " 'cdk10',\n",
       " 'interact',\n",
       " 'human',\n",
       " 'cyclin',\n",
       " 'c',\n",
       " 'longest',\n",
       " 'cdk10',\n",
       " 'isoform',\n",
       " 'p1',\n",
       " 'express',\n",
       " 'bait',\n",
       " 'protein',\n",
       " 'produc',\n",
       " 'strong',\n",
       " 'interact',\n",
       " 'phenotyp',\n",
       " 'full-length',\n",
       " 'cyclin',\n",
       " 'express',\n",
       " 'prey',\n",
       " 'protein',\n",
       " 'detect',\n",
       " 'phenotyp',\n",
       " 'cyclin',\n",
       " 'd1',\n",
       " 'p21',\n",
       " 'cip1',\n",
       " 'cdi1',\n",
       " 'kap',\n",
       " 'known',\n",
       " 'bind',\n",
       " 'partner',\n",
       " 'cdk',\n",
       " 'cdk1',\n",
       " 'cdk3',\n",
       " 'also',\n",
       " 'produc',\n",
       " 'y2h',\n",
       " 'signal',\n",
       " 'cyclin',\n",
       " 'albeit',\n",
       " 'notabl',\n",
       " 'weaker',\n",
       " 'observ',\n",
       " 'cdk10',\n",
       " 'interact',\n",
       " 'phenotyp',\n",
       " 'also',\n",
       " 'observ',\n",
       " 'full-length',\n",
       " 'cyclin',\n",
       " 'cdk10',\n",
       " 'protein',\n",
       " 'express',\n",
       " 'bait',\n",
       " 'prey',\n",
       " 'respect',\n",
       " 'test',\n",
       " 'differ',\n",
       " 'isoform',\n",
       " 'cdk10',\n",
       " 'cyclin',\n",
       " 'origin',\n",
       " 'altern',\n",
       " 'gene',\n",
       " 'splice',\n",
       " 'two',\n",
       " 'truncat',\n",
       " 'cyclin',\n",
       " 'protein',\n",
       " 'correspond',\n",
       " 'hypothet',\n",
       " 'product',\n",
       " 'two',\n",
       " 'mutat',\n",
       " 'fam58a',\n",
       " 'gene',\n",
       " 'found',\n",
       " 'star',\n",
       " 'syndrom',\n",
       " 'patient',\n",
       " 'none',\n",
       " 'shorter',\n",
       " 'isoform',\n",
       " 'produc',\n",
       " 'interact',\n",
       " 'phenotyp',\n",
       " 'c',\n",
       " 'new',\n",
       " 'window',\n",
       " 'download',\n",
       " 'ppt',\n",
       " 'cdk10',\n",
       " 'cyclin',\n",
       " 'form',\n",
       " 'interact',\n",
       " 'complex',\n",
       " 'schemat',\n",
       " 'represent',\n",
       " 'differ',\n",
       " 'protein',\n",
       " 'isoform',\n",
       " 'analyz',\n",
       " 'y2h',\n",
       " 'assay',\n",
       " 'amino',\n",
       " 'acid',\n",
       " 'number',\n",
       " 'indic',\n",
       " 'black',\n",
       " 'box',\n",
       " 'indic',\n",
       " 'intern',\n",
       " 'delet',\n",
       " 'red',\n",
       " 'box',\n",
       " 'indic',\n",
       " 'differ',\n",
       " 'amino',\n",
       " 'acid',\n",
       " 'sequenc',\n",
       " 'compar',\n",
       " 'cdk10',\n",
       " 'p1',\n",
       " 'y2h',\n",
       " 'assay',\n",
       " 'set',\n",
       " 'cdk',\n",
       " 'protein',\n",
       " 'express',\n",
       " 'bait',\n",
       " 'fusion',\n",
       " 'lexa',\n",
       " 'dna',\n",
       " 'bind',\n",
       " 'domain',\n",
       " 'cdk',\n",
       " 'interact',\n",
       " 'protein',\n",
       " 'express',\n",
       " 'prey',\n",
       " 'fusion',\n",
       " 'b42',\n",
       " 'transcript',\n",
       " 'activ',\n",
       " 'peg202',\n",
       " 'pjg4-5',\n",
       " 'empti',\n",
       " 'bait',\n",
       " 'prey',\n",
       " 'plasmid',\n",
       " 'express',\n",
       " 'lexa',\n",
       " 'b42',\n",
       " 'respect',\n",
       " 'lacz',\n",
       " 'use',\n",
       " 'report',\n",
       " 'gene',\n",
       " 'blue',\n",
       " 'yeast',\n",
       " 'indic',\n",
       " 'y2h',\n",
       " 'interact',\n",
       " 'phenotyp',\n",
       " 'y2h',\n",
       " 'assay',\n",
       " 'differ',\n",
       " 'cdk10',\n",
       " 'cyclin',\n",
       " 'isoform',\n",
       " 'amino-termin',\n",
       " 'region',\n",
       " 'ets2',\n",
       " 'known',\n",
       " 'interact',\n",
       " 'cdk10',\n",
       " 'also',\n",
       " 'assay',\n",
       " 'western',\n",
       " 'blot',\n",
       " 'analysi',\n",
       " 'myc-cdk10',\n",
       " 'wt',\n",
       " 'kd',\n",
       " 'cycm-v5-6hi',\n",
       " 'express',\n",
       " 'level',\n",
       " 'transfect',\n",
       " 'hek293',\n",
       " 'cell',\n",
       " 'western',\n",
       " 'blot',\n",
       " 'analysi',\n",
       " 'myc-cdk10',\n",
       " 'wt',\n",
       " 'kd',\n",
       " 'immunoprecipit',\n",
       " 'obtain',\n",
       " 'use',\n",
       " 'anti-myc',\n",
       " 'antibodi',\n",
       " 'input',\n",
       " 'correspond',\n",
       " 'g',\n",
       " 'total',\n",
       " 'lysat',\n",
       " 'obtain',\n",
       " 'hek293',\n",
       " 'cell',\n",
       " 'coexpress',\n",
       " 'myc-cdk10',\n",
       " 'wt',\n",
       " 'kd',\n",
       " 'cycm-v5-6hi',\n",
       " 'western',\n",
       " 'blot',\n",
       " 'analysi',\n",
       " 'immunoprecipit',\n",
       " 'obtain',\n",
       " 'use',\n",
       " 'anti-cdk10',\n",
       " 'antibodi',\n",
       " 'control',\n",
       " 'goat',\n",
       " 'antibodi',\n",
       " 'human',\n",
       " 'breast',\n",
       " 'cancer',\n",
       " 'mcf7',\n",
       " 'cell',\n",
       " 'input',\n",
       " 'correspond',\n",
       " 'g',\n",
       " 'mcf7',\n",
       " 'total',\n",
       " 'cell',\n",
       " 'lysat',\n",
       " 'lower',\n",
       " 'band',\n",
       " 'doublet',\n",
       " 'observ',\n",
       " 'upper',\n",
       " 'panel',\n",
       " 'comigr',\n",
       " 'exogen',\n",
       " 'express',\n",
       " 'untag',\n",
       " 'cdk10',\n",
       " 'thu',\n",
       " 'correspond',\n",
       " 'endogen',\n",
       " 'cdk10',\n",
       " 'upper',\n",
       " 'band',\n",
       " 'doublet',\n",
       " 'correspond',\n",
       " 'nonspecif',\n",
       " 'signal',\n",
       " 'demonstr',\n",
       " 'insensit',\n",
       " 'either',\n",
       " 'overexpress',\n",
       " 'cdk10',\n",
       " 'seen',\n",
       " 'left',\n",
       " 'lane',\n",
       " 'silenc',\n",
       " 'cdk10',\n",
       " 'anoth',\n",
       " 'experi',\n",
       " 'longer',\n",
       " 'gel',\n",
       " 'migrat',\n",
       " 'shown',\n",
       " 'next',\n",
       " 'examin',\n",
       " 'abil',\n",
       " 'cdk10',\n",
       " 'cyclin',\n",
       " 'interact',\n",
       " 'express',\n",
       " 'human',\n",
       " 'cell',\n",
       " 'e',\n",
       " 'test',\n",
       " 'wild-typ',\n",
       " 'cdk10',\n",
       " 'wt',\n",
       " 'kinas',\n",
       " 'dead',\n",
       " 'kd',\n",
       " 'mutant',\n",
       " 'bear',\n",
       " 'd181a',\n",
       " 'amino',\n",
       " 'acid',\n",
       " 'substitut',\n",
       " 'abolish',\n",
       " 'atp',\n",
       " 'bind',\n",
       " 'express',\n",
       " 'cyclin',\n",
       " 'm-v5-6hi',\n",
       " 'myc-cdk10',\n",
       " 'wt',\n",
       " 'kd',\n",
       " 'human',\n",
       " 'embryon',\n",
       " 'kidney',\n",
       " 'cell',\n",
       " 'line',\n",
       " 'hek293',\n",
       " 'express',\n",
       " 'level',\n",
       " 'cyclin',\n",
       " 'm-v5-6hi',\n",
       " 'significantli',\n",
       " 'increas',\n",
       " 'upon',\n",
       " 'coexpress',\n",
       " 'myc-cdk10',\n",
       " 'wt',\n",
       " 'kd',\n",
       " 'lesser',\n",
       " 'extent',\n",
       " 'myc-cdk10',\n",
       " 'wt',\n",
       " 'kd',\n",
       " 'increas',\n",
       " 'upon',\n",
       " 'coexpress',\n",
       " 'cyclin',\n",
       " 'm-v5-6hi',\n",
       " 'immunoprecipit',\n",
       " 'myc-cdk10',\n",
       " 'protein',\n",
       " 'detect',\n",
       " 'presenc',\n",
       " 'cyclin',\n",
       " 'cdk10',\n",
       " 'wt',\n",
       " 'kd',\n",
       " 'immunoprecipit',\n",
       " 'protein',\n",
       " 'coexpress',\n",
       " 'pair-wis',\n",
       " 'confirm',\n",
       " 'observ',\n",
       " 'detect',\n",
       " 'presenc',\n",
       " 'myc-cdk10',\n",
       " 'cyclin',\n",
       " 'm-v5-6hi',\n",
       " 'immunoprecipit',\n",
       " 'experi',\n",
       " 'confirm',\n",
       " 'lack',\n",
       " 'robust',\n",
       " 'interact',\n",
       " 'cdk10',\n",
       " 'p2',\n",
       " 'isoform',\n",
       " 'cyclin',\n",
       " 'detect',\n",
       " 'interact',\n",
       " 'endogen',\n",
       " 'protein',\n",
       " 'perform',\n",
       " 'immunoprecipit',\n",
       " 'nontransfect',\n",
       " 'mcf7',\n",
       " 'cell',\n",
       " 'deriv',\n",
       " 'human',\n",
       " 'breast',\n",
       " 'cancer',\n",
       " 'cdk10',\n",
       " 'cyclin',\n",
       " 'antibodi',\n",
       " 'detect',\n",
       " 'cognat',\n",
       " 'endogen',\n",
       " 'protein',\n",
       " 'western',\n",
       " 'blot',\n",
       " 'readili',\n",
       " 'detect',\n",
       " 'cyclin',\n",
       " 'immunoprecipit',\n",
       " 'obtain',\n",
       " 'cdk10',\n",
       " 'antibodi',\n",
       " 'control',\n",
       " 'antibodi',\n",
       " 'result',\n",
       " 'confirm',\n",
       " 'physic',\n",
       " 'interact',\n",
       " 'cdk10',\n",
       " 'cyclin',\n",
       " 'human',\n",
       " 'cell',\n",
       " 'unveil',\n",
       " 'hypothes',\n",
       " 'cdk10',\n",
       " 'cyclin',\n",
       " 'protein',\n",
       " 'kinas',\n",
       " 'activ',\n",
       " 'produc',\n",
       " 'gst-cdk10',\n",
       " 'strepii-cyclin',\n",
       " 'fusion',\n",
       " 'protein',\n",
       " 'insect',\n",
       " 'cell',\n",
       " 'either',\n",
       " 'individu',\n",
       " 'combin',\n",
       " 'observ',\n",
       " 'gst-cdk10',\n",
       " 'strepii-cyclin',\n",
       " 'copurifi',\n",
       " 'thu',\n",
       " 'confirm',\n",
       " 'interact',\n",
       " 'yet',\n",
       " 'anoth',\n",
       " 'cellular',\n",
       " 'model',\n",
       " 'perform',\n",
       " 'vitro',\n",
       " 'kinas',\n",
       " 'assay',\n",
       " 'purifi',\n",
       " 'protein',\n",
       " 'use',\n",
       " 'histon',\n",
       " 'h1',\n",
       " 'gener',\n",
       " 'substrat',\n",
       " 'histon',\n",
       " 'h1',\n",
       " 'phosphoryl',\n",
       " 'detect',\n",
       " 'lysat',\n",
       " 'cell',\n",
       " 'coexpress',\n",
       " 'gst-cdk10',\n",
       " 'strepii-cyclin',\n",
       " 'phosphoryl',\n",
       " 'detect',\n",
       " 'gst-cdk10',\n",
       " 'strepii-cyclin',\n",
       " 'express',\n",
       " 'alon',\n",
       " 'strepii-cyclin',\n",
       " 'coexpress',\n",
       " 'gst-cdk10',\n",
       " 'kd',\n",
       " 'next',\n",
       " 'investig',\n",
       " 'whether',\n",
       " 'ets2',\n",
       " 'known',\n",
       " 'interact',\n",
       " 'cdk10',\n",
       " 'phosphoryl',\n",
       " 'substrat',\n",
       " 'cdk10',\n",
       " 'cyclin',\n",
       " 'detect',\n",
       " 'strong',\n",
       " 'phosphoryl',\n",
       " 'ets2',\n",
       " 'gst-cdk10',\n",
       " 'strepii-cyclin',\n",
       " 'purifi',\n",
       " 'heterodim',\n",
       " 'wherea',\n",
       " 'phosphoryl',\n",
       " 'detect',\n",
       " 'use',\n",
       " 'gst-cdk10',\n",
       " 'alon',\n",
       " 'gst-cdk10',\n",
       " 'kd',\n",
       " 'strepii-cyclin',\n",
       " 'heterodim',\n",
       " 'new',\n",
       " 'window',\n",
       " 'download',\n",
       " 'ppt',\n",
       " 'cdk10',\n",
       " 'cyclin',\n",
       " 'm-depend',\n",
       " 'protein',\n",
       " 'kinas',\n",
       " 'vitro',\n",
       " 'protein',\n",
       " 'kinas',\n",
       " 'assay',\n",
       " 'histon',\n",
       " 'h1',\n",
       " 'lysat',\n",
       " 'insect',\n",
       " 'cell',\n",
       " 'express',\n",
       " 'differ',\n",
       " 'protein',\n",
       " 'purifi',\n",
       " 'glutathion',\n",
       " 'sepharos',\n",
       " 'matrix',\n",
       " 'captur',\n",
       " 'gst-cdk10',\n",
       " 'wt',\n",
       " 'kd',\n",
       " 'fusion',\n",
       " 'protein',\n",
       " 'alon',\n",
       " 'complex',\n",
       " 'str-cycm',\n",
       " 'fusion',\n",
       " 'protein',\n",
       " 'purifi',\n",
       " 'protein',\n",
       " 'express',\n",
       " 'level',\n",
       " 'analyz',\n",
       " 'western',\n",
       " 'blot',\n",
       " 'top',\n",
       " 'upper',\n",
       " 'middl',\n",
       " 'kinas',\n",
       " 'activ',\n",
       " 'determin',\n",
       " 'autoradiographi',\n",
       " 'histon',\n",
       " 'h1',\n",
       " 'whose',\n",
       " 'ad',\n",
       " 'amount',\n",
       " 'visual',\n",
       " 'coomassi',\n",
       " 'stain',\n",
       " 'lower',\n",
       " 'middl',\n",
       " 'bottom',\n",
       " 'use',\n",
       " 'purifi',\n",
       " 'recombin',\n",
       " '6his-ets2',\n",
       " 'substrat',\n",
       " 'cdk10',\n",
       " 'silenc',\n",
       " 'shown',\n",
       " 'increas',\n",
       " 'ets2-driven',\n",
       " 'c-raf',\n",
       " 'transcript',\n",
       " 'activ',\n",
       " 'mapk',\n",
       " 'pathway',\n",
       " 'investig',\n",
       " 'whether',\n",
       " 'cyclin',\n",
       " 'also',\n",
       " 'involv',\n",
       " 'regulatori',\n",
       " 'pathway',\n",
       " 'aim',\n",
       " 'highli',\n",
       " 'specif',\n",
       " 'silenc',\n",
       " 'use',\n",
       " 'sirna',\n",
       " 'pool',\n",
       " 'mix',\n",
       " 'four',\n",
       " 'differ',\n",
       " 'sirna',\n",
       " 'low',\n",
       " 'final',\n",
       " 'concentr',\n",
       " '10',\n",
       " 'nm',\n",
       " 'cdk10',\n",
       " 'cyclin',\n",
       " 'sirna',\n",
       " 'pool',\n",
       " 'silenc',\n",
       " 'express',\n",
       " 'cognat',\n",
       " 'target',\n",
       " 'c',\n",
       " 'interestingli',\n",
       " 'cyclin',\n",
       " 'sirna',\n",
       " 'pool',\n",
       " 'also',\n",
       " 'caus',\n",
       " 'mark',\n",
       " 'decreas',\n",
       " 'cdk10',\n",
       " 'protein',\n",
       " 'level',\n",
       " 'result',\n",
       " 'shown',\n",
       " 'suggest',\n",
       " 'cyclin',\n",
       " 'bind',\n",
       " 'stabil',\n",
       " 'cdk10',\n",
       " 'cyclin',\n",
       " 'silenc',\n",
       " 'induc',\n",
       " 'increas',\n",
       " 'c-raf',\n",
       " 'protein',\n",
       " 'mrna',\n",
       " 'level',\n",
       " 'c',\n",
       " ...]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "port_tokenizer(df_processed[0])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is taking forever, Lets try possibly using the hashing vectorizer and feeding that into logsitic regression.   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Michael\\Anaconda3\\envs\\tensorflow\\lib\\site-packages\\sklearn\\utils\\deprecation.py:70: DeprecationWarning: Function transform is deprecated; Support to use estimators as feature selectors will be removed in version 0.19. Use SelectFromModel instead.\n",
      "  warnings.warn(msg, category=DeprecationWarning)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0.68999999999999995"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn.feature_extraction.text import HashingVectorizer\n",
    "tfidf_vect = HashingVectorizer(tokenizer=port_tokenizer)\n",
    "# pipe = Pipeline([('tfidf',tfidf_vect),('log_reg',LogisticRegression())])\n",
    "# cross_val_score(pipe,df_processed.values[:10],y[:10],n_jobs=-1,cv=5)\n",
    "a = tfidf_vect.fit_transform(df_processed.values[:300])\n",
    "lr = LogisticRegression()\n",
    "lr.fit_transform(a[:200],y[:200])\n",
    "lr.score(a[200:300],y[200:300])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from sklearn.linear_model import SGDClassifier"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "unsupported format string passed to csr_matrix.__format__",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-77-0936e585a2cf>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[0;32m     17\u001b[0m     \u001b[0ms\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpartial_fit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mx\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0my\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mclasses\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0munique\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdf_variants\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mClass\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     18\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[0mi\u001b[0m\u001b[1;33m%\u001b[0m\u001b[1;36m5\u001b[0m\u001b[1;33m==\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 19\u001b[1;33m         \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'iteration: {:d}\\n\\ttraining accuracy: {:.3f}\\n\\ttest accuracy: {:.3f}'\u001b[0m\u001b[1;33m.\u001b[0m             \u001b[0mformat\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mx_ts\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0my_ts\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     20\u001b[0m \u001b[0ms\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mscore\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mtfidf_vect\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtransform\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mX_ts\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0my_ts\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mTypeError\u001b[0m: unsupported format string passed to csr_matrix.__format__"
     ]
    }
   ],
   "source": [
    "from sklearn.utils import shuffle\n",
    "from sklearn.model_selection import train_test_split\n",
    "n_batches=100\n",
    "batch_size=100\n",
    "\n",
    "s = SGDClassifier()\n",
    "X_train,X_test,y_train,y_test = train_test_split(df_processed,df_variants.Class)\n",
    "max_batches = X_train.shape[0]//batch_size\n",
    "tfidf_vect = HashingVectorizer(tokenizer=port_tokenizer)\n",
    "x_ts,y_ts = tfidf_vect.transform(X_test[:500]),y_test[:500]\n",
    "for i in range(n_batches):\n",
    "    if i%max_batches==0:\n",
    "        X_train,y_train = shuffle(X_train,y_train)\n",
    "    x = X_train[batch_size*(i%max_batches):batch_size*((i%max_batches)+1)]\n",
    "    x = tfidf_vect.transform(x)\n",
    "    y = y_train[batch_size*(i%max_batches):batch_size*((i%max_batches)+1)]\n",
    "    s.partial_fit(x,y,classes = np.unique(df_variants.Class))\n",
    "    if i%5==0:\n",
    "        print('iteration: {:d}\\n\\ttraining accuracy: {:.3f}\\n\\ttest accuracy: {:.3f}'.\\\n",
    "             format(i,s.score(x,y),s.score(x_ts,y_ts)))\n",
    "s.score(tfidf_vect.transform(X_ts),y_ts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 0\n",
      "\ttraining loss: 0.876\n",
      "\ttest loss: 2.629\n",
      "iteration: 5\n",
      "\ttraining loss: 0.437\n",
      "\ttest loss: 1.822\n",
      "iteration: 10\n",
      "\ttraining loss: 0.563\n",
      "\ttest loss: 1.544\n",
      "iteration: 15\n",
      "\ttraining loss: 0.515\n",
      "\ttest loss: 1.314\n",
      "iteration: 20\n",
      "\ttraining loss: 0.655\n",
      "\ttest loss: 1.222\n",
      "iteration: 25\n",
      "\ttraining loss: 0.591\n",
      "\ttest loss: 1.128\n",
      "iteration: 30\n",
      "\ttraining loss: 0.572\n",
      "\ttest loss: 1.175\n",
      "iteration: 35\n",
      "\ttraining loss: 0.616\n",
      "\ttest loss: 1.139\n",
      "iteration: 40\n",
      "\ttraining loss: 0.634\n",
      "\ttest loss: 1.082\n",
      "iteration: 45\n",
      "\ttraining loss: 0.743\n",
      "\ttest loss: 1.080\n"
     ]
    }
   ],
   "source": [
    "from sklearn.utils import shuffle\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.feature_extraction.text import TfidfTransformer\n",
    "n_batches=50\n",
    "batch_size=100\n",
    "from sklearn.metrics import log_loss\n",
    "s = SGDClassifier(loss='log')\n",
    "for i in range(0,n_batches):\n",
    "    if i%max_batches==0:\n",
    "        X_train,y_train = shuffle(X_train,y_train)\n",
    "    x = X_train[batch_size*(i%max_batches):batch_size*((i%max_batches)+1)]\n",
    "    x = tfidf_vect.transform(x)\n",
    "    y = y_train[batch_size*(i%max_batches):batch_size*((i%max_batches)+1)]\n",
    "    s.partial_fit(x,y,classes = np.unique(df_variants.Class))\n",
    "    if i%5==0:\n",
    "        y_pred = s.predict_proba(x)\n",
    "        y_ts_pred = s.predict_proba(x_ts)\n",
    "        print('iteration: {:d}\\n\\ttraining loss: {:.3f}\\n\\ttest loss: {:.3f}'.\\\n",
    "             format(i,log_loss(y,s.predict_proba(x),labels=range(1,10)),\n",
    "                    log_loss(y_ts,s.predict_proba(x_ts),labels=range(1,10))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "iteration: 0\n",
      "\ttraining loss: 0.575\n",
      "\ttest loss: 1.082\n",
      "iteration: 5\n",
      "\ttraining loss: 0.680\n",
      "\ttest loss: 1.051\n",
      "iteration: 10\n",
      "\ttraining loss: 0.684\n",
      "\ttest loss: 1.056\n",
      "iteration: 15\n",
      "\ttraining loss: 0.628\n",
      "\ttest loss: 1.071\n",
      "iteration: 20\n",
      "\ttraining loss: 0.715\n",
      "\ttest loss: 1.038\n",
      "iteration: 25\n",
      "\ttraining loss: 0.637\n",
      "\ttest loss: 1.059\n",
      "iteration: 30\n",
      "\ttraining loss: 0.761\n",
      "\ttest loss: 1.068\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "KeyboardInterrupt\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for i in range(0,n_batches):\n",
    "    if i%max_batches==0:\n",
    "        X_train,y_train = shuffle(X_train,y_train)\n",
    "    x = X_train[batch_size*(i%max_batches):batch_size*((i%max_batches)+1)]\n",
    "    x = tfidf_vect.transform(x)\n",
    "    y = y_train[batch_size*(i%max_batches):batch_size*((i%max_batches)+1)]\n",
    "    s.partial_fit(x,y,classes = np.unique(df_variants.Class))\n",
    "    if i%5==0:\n",
    "        y_pred = s.predict_proba(x)\n",
    "        y_ts_pred = s.predict_proba(x_ts)\n",
    "        print('iteration: {:d}\\n\\ttraining loss: {:.3f}\\n\\ttest loss: {:.3f}'.\\\n",
    "             format(i,log_loss(y,s.predict_proba(x),labels=range(1,10)),\n",
    "                    log_loss(y_ts,s.predict_proba(x_ts),labels=range(1,10))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "1+1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "df = pd.read_csv('../data/training_variants.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df.to_pickle()"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [conda env:tensorflow]",
   "language": "python",
   "name": "conda-env-tensorflow-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
